// @lucca-front/ng/esm2022/form-field/input.directive.mjs
import { Directive, ElementRef, HostBinding, inject } from '@angular/core';
import * as i0 from "@angular/core";
class InputDirective {
    constructor() {
        this.clazz = 'textField-input-value';
        this.host = inject(ElementRef);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: InputDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: InputDirective, isStandalone: true, selector: "[luInput]", host: { properties: { "class": "this.clazz" } }, ngImport: i0 }); }
}
export { InputDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: InputDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luInput]',
                    standalone: true,
                }]
        }], propDecorators: { clazz: [{
                type: HostBinding,
                args: ['class']
            }] } });
