// @lucca-front/ng/esm2022/option/selector/all/tree-select-all.component.mjs
import { ChangeDetectionStrategy, Component, forwardRef, HostBinding } from '@angular/core';
import { getIntl } from '@lucca-front/ng/core';
import { Subject } from 'rxjs';
import { tap } from 'rxjs/operators';
import { ALuTreeOptionOperator } from '../../operator/index';
import { ALuTreeOptionSelector } from '../tree-option-selector.model';
import { LU_OPTION_SELECT_ALL_TRANSLATIONS } from './select-all.translate';
import * as i0 from "@angular/core";
class LuTreeOptionSelectAllComponent extends ALuTreeOptionOperator {
    constructor() {
        super(...arguments);
        this.multiple = true;
        this.onSelectValue = new Subject();
        this.fixed = true;
        this.intl = getIntl(LU_OPTION_SELECT_ALL_TRANSLATIONS);
    }
    set inOptions$(in$) {
        this.outOptions$ = in$.pipe(tap((options) => (this.flatOptions = this.flattenTree(options))));
    }
    selectAll() {
        this.onSelectValue.next([...this.flatOptions]);
    }
    deselectAll() {
        this.onSelectValue.next([]);
    }
    setValue(values) {
        this._values = values;
    }
    flattenTree(tree = []) {
        return tree.map((t) => [t.value, ...this.flattenTree(t.children)]).reduce((agg, cur) => [...agg, ...cur], []);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTreeOptionSelectAllComponent, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuTreeOptionSelectAllComponent, isStandalone: true, selector: "lu-tree-option-select-all", host: { properties: { "class.position-fixed": "this.fixed" } }, providers: [
            {
                provide: ALuTreeOptionOperator,
                useExisting: forwardRef(() => LuTreeOptionSelectAllComponent),
                multi: true,
            },
            {
                provide: ALuTreeOptionSelector,
                useExisting: forwardRef(() => LuTreeOptionSelectAllComponent),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0, template: "<button type=\"button\" (click)=\"selectAll()\" class=\"link selectAll-button\">{{ intl.select }}</button>\n<button type=\"button\" (click)=\"deselectAll()\" class=\"link selectAll-button u-marginLeftXS\">{{ intl.deselect }}</button>\n", styles: [":root{--components-options-item-padding-vertical: var(--spacings-XS);--components-options-item-padding-horizontal: var(--spacings-XS);--components-options-item-multiple-padding: 2.25rem;--components-options-item-icon-color: var(--palettes-grey-800);--components-options-checkbox-left: .5rem;--components-options-checkbox-size: 1.25rem;--components-options-checkbox-color: var(--palettes-primary-700);--components-options-checkbox-border-radius: 6px;--components-options-checkbox-border-color: var(--palettes-grey-700);--components-options-establishment-multiple-padding: 2rem}:host .selectAll-button{color:var(--palettes-grey-600);font-size:var(--sizes-S-fontSize);outline:none}:host .selectAll-button:hover{color:var(--palettes-grey-800)}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuTreeOptionSelectAllComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTreeOptionSelectAllComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lu-tree-option-select-all', changeDetection: ChangeDetectionStrategy.OnPush, standalone: true, providers: [
                        {
                            provide: ALuTreeOptionOperator,
                            useExisting: forwardRef(() => LuTreeOptionSelectAllComponent),
                            multi: true,
                        },
                        {
                            provide: ALuTreeOptionSelector,
                            useExisting: forwardRef(() => LuTreeOptionSelectAllComponent),
                            multi: true,
                        },
                    ], template: "<button type=\"button\" (click)=\"selectAll()\" class=\"link selectAll-button\">{{ intl.select }}</button>\n<button type=\"button\" (click)=\"deselectAll()\" class=\"link selectAll-button u-marginLeftXS\">{{ intl.deselect }}</button>\n", styles: [":root{--components-options-item-padding-vertical: var(--spacings-XS);--components-options-item-padding-horizontal: var(--spacings-XS);--components-options-item-multiple-padding: 2.25rem;--components-options-item-icon-color: var(--palettes-grey-800);--components-options-checkbox-left: .5rem;--components-options-checkbox-size: 1.25rem;--components-options-checkbox-color: var(--palettes-primary-700);--components-options-checkbox-border-radius: 6px;--components-options-checkbox-border-color: var(--palettes-grey-700);--components-options-establishment-multiple-padding: 2rem}:host .selectAll-button{color:var(--palettes-grey-600);font-size:var(--sizes-S-fontSize);outline:none}:host .selectAll-button:hover{color:var(--palettes-grey-800)}\n"] }]
        }], propDecorators: { fixed: [{
                type: HostBinding,
                args: ['class.position-fixed']
            }] } });
