// @lucca-front/ng/esm2022/api/service/api-hybrid.service.mjs
import { Injectable } from '@angular/core';
import { ALuApiService } from './api-service.model';
import { LuApiV3Service } from './api-v3.service';
import { LuApiV4Service } from './api-v4.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
class LuApiHybridService extends ALuApiService {
    set standard(std) {
        this._standard = std;
    }
    // both
    set api(api) {
        this._v3Service.api = api;
        this._v4Service.api = api;
    }
    set filters(filters) {
        this._v3Service.filters = filters;
        this._v4Service.filters = filters;
    }
    // api v3 only
    set fields(fields) {
        this._v3Service.fields = fields;
    }
    set orderBy(orderBy) {
        this._v3Service.orderBy = orderBy;
    }
    // api v4 only
    set sort(sort) {
        this._v4Service.sort = sort;
    }
    get _service() {
        switch (this._standard) {
            case 'v4':
                return this._v4Service;
            case 'v3':
            default:
                return this._v3Service;
        }
    }
    constructor(_http) {
        super();
        this._http = _http;
        this._standard = 'v3';
        this._v3Service = new LuApiV3Service(this._http);
        this._v4Service = new LuApiV4Service(this._http);
    }
    getAll(filters = []) {
        return this._service.getAll(filters);
    }
    getPaged(page, filters = []) {
        return this._service.getPaged(page, filters);
    }
    searchAll(clue, filters = []) {
        return this._service.searchAll(clue, filters);
    }
    searchPaged(clue, page, filters = []) {
        return this._service.searchPaged(clue, page, filters);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuApiHybridService, deps: [{ token: i1.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuApiHybridService }); }
}
export { LuApiHybridService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuApiHybridService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.HttpClient }]; } });
