// @lucca-front/ng/esm2022/api/service/api-v3.service.mjs
import { Injectable } from '@angular/core';
import { map } from 'rxjs/operators';
import { ALuApiService } from './api-service.model';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
const MAGIC_PAGE_SIZE = 20;
class LuApiV3Service extends ALuApiService {
    set api(api) {
        this._api = api;
    }
    set fields(fields) {
        if (fields) {
            this._fields = `fields=${fields}`;
        }
    }
    set filters(filters) {
        if (filters) {
            this._filters = filters || [];
        }
    }
    set orderBy(orderBy) {
        if (orderBy) {
            this._orderBy = `orderBy=${orderBy}`;
        }
    }
    get url() {
        return `${this._api}?${[...this._filters, this._orderBy, this._fields].filter((f) => !!f).join('&')}`;
    }
    constructor(_http) {
        super();
        this._http = _http;
        this._fields = 'fields=id,name';
        this._filters = [];
        this._orderBy = 'orderBy=name,asc';
    }
    getAll(filters = []) {
        return this._get([this.url, ...filters].join('&'));
    }
    getPaged(page, filters = []) {
        const paging = `paging=${page * MAGIC_PAGE_SIZE},${MAGIC_PAGE_SIZE}`;
        const url = [this.url, paging, ...filters].join('&');
        return this._get(url);
    }
    searchAll(clue, filters = []) {
        if (!clue) {
            return this.getAll(filters);
        }
        const url = [this.url, this._clueFilter(clue), ...filters].join('&');
        return this._get(url);
    }
    searchPaged(clue, page, filters = []) {
        if (!clue) {
            return this.getPaged(page, filters);
        }
        const paging = `paging=${page * MAGIC_PAGE_SIZE},${MAGIC_PAGE_SIZE}`;
        const url = [this.url, this._clueFilter(clue), paging, ...filters].join('&');
        return this._get(url);
    }
    _get(url) {
        return this._http.get(url).pipe(map((response) => response.data.items));
    }
    _clueFilter(clue) {
        const urlSafeClue = encodeURIComponent(clue);
        return `name=like,${urlSafeClue}`;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuApiV3Service, deps: [{ token: i1.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuApiV3Service }); }
}
export { LuApiV3Service };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuApiV3Service, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.HttpClient }]; } });
