// @lucca-front/ng/esm2022/number/number.pipe.mjs
import { formatNumber, getLocaleNumberSymbol, NumberSymbol } from '@angular/common';
import { Inject, LOCALE_ID, Pipe } from '@angular/core';
import * as i0 from "@angular/core";
class LuNumberPipe {
    constructor(locale) {
        this.locale = locale;
    }
    transform(number, precision = 2) {
        const formatted = formatNumber(number, this.locale, `1.${precision}-${precision}`);
        const separator = getLocaleNumberSymbol(this.locale, NumberSymbol.Decimal);
        const split = formatted.split(separator);
        const integral = split[0];
        const decimal = split[1];
        const hideDecimal = Math.round(number) === number;
        if (precision > 0) {
            return `${integral}<span class="decimal-part${hideDecimal ? ' u-hidden' : ''}">${separator}${decimal}</span>`;
        }
        else {
            return integral;
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberPipe, deps: [{ token: LOCALE_ID }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuNumberPipe, isStandalone: true, name: "luNumber" }); }
}
export { LuNumberPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'luNumber',
                    pure: true,
                    standalone: true,
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [LOCALE_ID]
                }] }]; } });
