// @lucca-front/ng/esm2022/simple-select/input/panel-ref.factory.mjs
import { Overlay, OverlayPositionBuilder, ScrollStrategyOptions } from '@angular/cdk/overlay';
import { ComponentPortal } from '@angular/cdk/portal';
import { ElementRef, inject, Injectable, Injector } from '@angular/core';
import { LuSelectPanelRef } from '@lucca-front/ng/core-select';
import { takeUntil } from 'rxjs';
import { LuSelectPanelComponent } from '../panel';
import { SIMPLE_SELECT_PANEL_DATA } from '../select.model';
import * as i0 from "@angular/core";
class SelectPanelRef extends LuSelectPanelRef {
    constructor(overlayRef, parentInjector, panelData) {
        super();
        this.overlayRef = overlayRef;
        const injector = Injector.create({
            providers: [
                { provide: LuSelectPanelRef, useValue: this },
                { provide: SIMPLE_SELECT_PANEL_DATA, useValue: panelData },
            ],
            parent: parentInjector,
        });
        this.portalRef = new ComponentPortal(LuSelectPanelComponent, undefined, injector);
        this.panelRef = overlayRef.attach(this.portalRef);
        this.instance = this.panelRef.instance;
        overlayRef
            .backdropClick()
            .pipe(takeUntil(this.closed))
            .subscribe(() => this.close());
    }
    emitValue(value) {
        this.valueChanged.emit(value);
        this.close();
    }
    close() {
        super.close();
        this.panelRef.destroy();
        this.overlayRef.detach();
    }
}
class LuSimpleSelectPanelRefFactory {
    constructor() {
        this.overlay = inject(Overlay);
        this.elementRef = inject(ElementRef);
        this.positionBuilder = inject(OverlayPositionBuilder);
        this.scrollStrategies = inject(ScrollStrategyOptions);
        this.parentInjector = inject(Injector);
    }
    buildPanelRef(panelData, overlayConfigOverride = {}) {
        const overlayConfig = this.buildOverlayConfig(overlayConfigOverride);
        const overlayRef = this.overlay.create(overlayConfig);
        return new SelectPanelRef(overlayRef, this.parentInjector, panelData);
    }
    buildOverlayConfig(overlayConfigOverride = {}) {
        const overlayConfig = overlayConfigOverride || {};
        overlayConfig.positionStrategy = this.positionBuilder.flexibleConnectedTo(this.elementRef).withPositions([
            {
                originX: 'start',
                originY: 'bottom',
                overlayX: 'start',
                overlayY: 'top',
            },
            {
                originX: 'end',
                originY: 'bottom',
                overlayX: 'end',
                overlayY: 'top',
            },
            {
                originX: 'start',
                originY: 'top',
                overlayX: 'start',
                overlayY: 'bottom',
            },
            {
                originX: 'end',
                originY: 'top',
                overlayX: 'end',
                overlayY: 'bottom',
            },
        ]);
        overlayConfig.scrollStrategy = this.scrollStrategies.reposition();
        overlayConfig.minWidth = this.elementRef.nativeElement.clientWidth;
        overlayConfig.maxHeight = '100vh';
        overlayConfig.maxWidth = '100vw';
        return overlayConfig;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectPanelRefFactory, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectPanelRefFactory }); }
}
export { LuSimpleSelectPanelRefFactory };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectPanelRefFactory, decorators: [{
            type: Injectable
        }] });
