// @lucca-front/ng/esm2022/user/display/display-format.model.mjs
import { InjectionToken } from '@angular/core';
export var LuDisplayFullname;
(function (LuDisplayFullname) {
    LuDisplayFullname["firstlast"] = "fl";
    LuDisplayFullname["lastfirst"] = "lf";
    LuDisplayFullname["first"] = "f";
    LuDisplayFullname["last"] = "l";
})(LuDisplayFullname || (LuDisplayFullname = {}));
export var LuDisplayInitials;
(function (LuDisplayInitials) {
    LuDisplayInitials["firstlast"] = "FL";
    LuDisplayInitials["lastfirst"] = "LF";
    LuDisplayInitials["last"] = "L";
    LuDisplayInitials["first"] = "F";
})(LuDisplayInitials || (LuDisplayInitials = {}));
export var LuDisplayHybrid;
(function (LuDisplayHybrid) {
    LuDisplayHybrid["firstIlastFull"] = "Fl";
    LuDisplayHybrid["firstFulllastI"] = "fL";
    LuDisplayHybrid["lastIfirstFull"] = "Lf";
    LuDisplayHybrid["lastFullfirstI"] = "lF";
})(LuDisplayHybrid || (LuDisplayHybrid = {}));
/** Injection token that can be used to change the default displayed user format. */
export const LU_DEFAULT_DISPLAY_POLICY = new InjectionToken('LuDisplayFormat', { factory: () => LuDisplayFullname.lastfirst });
