// @lucca-front/ng/esm2022/user/display/user-display.module.mjs
import { NgModule } from '@angular/core';
import { LuUserDisplayPipe } from './user-display.pipe';
import * as i0 from "@angular/core";
class LuUserDisplayModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayModule, imports: [LuUserDisplayPipe], exports: [LuUserDisplayPipe] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayModule, providers: [LuUserDisplayPipe] }); }
}
export { LuUserDisplayModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [LuUserDisplayPipe],
                    providers: [LuUserDisplayPipe],
                    exports: [LuUserDisplayPipe],
                }]
        }] });
