// @lucca-front/ng/esm2022/user/display/user-display.pipe.mjs
import { inject, Pipe } from '@angular/core';
import { LuDisplayFullname, LuDisplayHybrid, LuDisplayInitials, LU_DEFAULT_DISPLAY_POLICY } from './display-format.model';
import * as i0 from "@angular/core";
/**
 * Displays a user name according to specified format. Supported formats: f for first name,
 * F for first initial, l for last name, L for last initial.
 */
export function luUserDisplay(user, format = LuDisplayFullname.lastfirst) {
    let result = '';
    if (!!user && !!user.firstName && !!user.lastName) {
        switch (format) {
            case LuDisplayFullname.lastfirst:
                result = user.lastName + ' ' + user.firstName;
                break;
            case LuDisplayFullname.firstlast:
                result = user.firstName + ' ' + user.lastName;
                break;
            case LuDisplayFullname.first:
                result = user.firstName;
                break;
            case LuDisplayFullname.last:
                result = user.lastName;
                break;
            case LuDisplayInitials.lastfirst:
                result = user.lastName.charAt(0) + user.firstName.charAt(0);
                break;
            case LuDisplayInitials.firstlast:
                result = user.firstName.charAt(0) + user.lastName.charAt(0);
                break;
            case LuDisplayInitials.first:
                result = user.firstName.charAt(0);
                break;
            case LuDisplayInitials.last:
                result = user.lastName.charAt(0);
                break;
            case LuDisplayHybrid.firstIlastFull:
                result = user.firstName.charAt(0) + '. ' + user.lastName;
                break;
            case LuDisplayHybrid.lastIfirstFull:
                result = user.lastName.charAt(0) + '. ' + user.firstName;
                break;
            case LuDisplayHybrid.lastFullfirstI:
                result = user.lastName + ' ' + user.firstName.charAt(0) + '.';
                break;
            case LuDisplayHybrid.firstFulllastI:
                result = user.firstName + ' ' + user.lastName.charAt(0) + '.';
                break;
            default:
                break;
        }
    }
    return result;
}
/**
 * Displays a user name according to specified format. Supported formats: f for first name,
 * F for first initial, l for last name, L for last initial.
 */
class LuUserDisplayPipe {
    constructor() {
        this.defaultFormat = inject(LU_DEFAULT_DISPLAY_POLICY);
    }
    transform(user, format = this.defaultFormat) {
        return luUserDisplay(user, format);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayPipe, isStandalone: true, name: "luUserDisplay" }); }
}
export { LuUserDisplayPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserDisplayPipe, decorators: [{
            type: Pipe,
            args: [{ name: 'luUserDisplay', standalone: true }]
        }] });
