// @lucca-front/ng/esm2022/option/operator/feeder/option-feeder.component.mjs
import { ChangeDetectionStrategy, Component, forwardRef, Input } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { ALuOptionOperator } from '../option-operator.model';
import * as i0 from "@angular/core";
class LuOptionFeederComponent {
    constructor() {
        this.outOptions$ = new BehaviorSubject([]);
    }
    set options(options) {
        this.outOptions$.next(options);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuOptionFeederComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuOptionFeederComponent, isStandalone: true, selector: "lu-option-feeder", inputs: { options: "options" }, providers: [
            {
                provide: ALuOptionOperator,
                useExisting: forwardRef(() => LuOptionFeederComponent),
                multi: true,
            },
        ], ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuOptionFeederComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuOptionFeederComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lu-option-feeder', template: '', changeDetection: ChangeDetectionStrategy.OnPush, standalone: true, providers: [
                        {
                            provide: ALuOptionOperator,
                            useExisting: forwardRef(() => LuOptionFeederComponent),
                            multi: true,
                        },
                    ] }]
        }], propDecorators: { options: [{
                type: Input
            }] } });
