// @lucca-front/ng/esm2022/toast/toasts.service.mjs
import { Injectable } from '@angular/core';
import { BehaviorSubject } from 'rxjs';
import { defaultToastDuration } from './toasts.model';
import * as i0 from "@angular/core";
class LuToastsService {
    constructor() {
        this.toasts$ = new BehaviorSubject([]);
    }
    addToast(toastInput) {
        const toast = this.getToast(toastInput);
        this.toasts$.next([...this.toasts$.value, toast]);
        return toast;
    }
    removeToast(toast) {
        const updatedToasts = this.toasts$.value.filter(({ id }) => id !== toast.id);
        this.toasts$.next(updatedToasts);
    }
    isOnlyDismissibleManually({ duration }) {
        return duration === null;
    }
    getToast(toastInput) {
        const id = this.generateId();
        const duration = this.isOnlyDismissibleManually(toastInput) ? toastInput.duration : toastInput.duration ?? defaultToastDuration;
        return { ...toastInput, id, duration };
    }
    generateId() {
        const randomString = Math.random().toString(36).substring(2, 9);
        return `_${randomString}`;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuToastsService, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuToastsService, providedIn: 'root' }); }
}
export { LuToastsService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuToastsService, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }] });
