// @lucca-front/ng/esm2022/option/operator/for-options/for-options.directive.mjs
import { NgForOf } from '@angular/common';
import { Directive, forwardRef, Input } from '@angular/core';
import { Subscription } from 'rxjs';
import { ALuOptionOperator } from '../option-operator.model';
import * as i0 from "@angular/core";
class LuForOptionsDirective extends NgForOf {
    set inOptions$(options$) {
        this._subs.add(options$.subscribe((options) => {
            this.ngForOf = options;
            this._changeDetectionRef.markForCheck();
        }));
        this.outOptions$ = options$;
    }
    set luForOptionsTrackBy(fn) {
        this.ngForTrackBy = fn;
    }
    constructor(_viewContainer, _template, _differs, _changeDetectionRef) {
        super(_viewContainer, _template, _differs);
        this._changeDetectionRef = _changeDetectionRef;
        this._subs = new Subscription();
    }
    ngOnDestroy() {
        this._subs.unsubscribe();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForOptionsDirective, deps: [{ token: i0.ViewContainerRef }, { token: i0.TemplateRef }, { token: i0.IterableDiffers }, { token: i0.ChangeDetectorRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuForOptionsDirective, isStandalone: true, selector: "[luForOptions]", inputs: { luForOptionsTrackBy: "luForOptionsTrackBy" }, providers: [
            {
                provide: ALuOptionOperator,
                useExisting: forwardRef(() => LuForOptionsDirective),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0 }); }
}
export { LuForOptionsDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForOptionsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luForOptions]',
                    standalone: true,
                    providers: [
                        {
                            provide: ALuOptionOperator,
                            useExisting: forwardRef(() => LuForOptionsDirective),
                            multi: true,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.TemplateRef }, { type: i0.IterableDiffers }, { type: i0.ChangeDetectorRef }]; }, propDecorators: { luForOptionsTrackBy: [{
                type: Input
            }] } });
