// @lucca-front/ng/esm2022/option/operator/for-options/for-tree-options.directive.mjs
import { Directive, forwardRef } from '@angular/core';
import { ALuTreeOptionOperator } from '../tree-option-operator.model';
import { LuForOptionsDirective } from './for-options.directive';
import * as i0 from "@angular/core";
class LuForTreeOptionsDirective extends LuForOptionsDirective {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForTreeOptionsDirective, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuForTreeOptionsDirective, isStandalone: true, selector: "[luForTreeOptions]", providers: [
            {
                provide: ALuTreeOptionOperator,
                useExisting: forwardRef(() => LuForTreeOptionsDirective),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0 }); }
}
export { LuForTreeOptionsDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForTreeOptionsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luForTreeOptions]',
                    standalone: true,
                    providers: [
                        {
                            provide: ALuTreeOptionOperator,
                            useExisting: forwardRef(() => LuForTreeOptionsDirective),
                            multi: true,
                        },
                    ],
                }]
        }] });
