// @lucca-front/ng/esm2022/api/select/pager/api-pager.model.mjs
import { merge, of, Subject } from 'rxjs';
import { catchError, distinctUntilChanged, map, switchMap, tap } from 'rxjs/operators';
var Strategy;
(function (Strategy) {
    Strategy[Strategy["append"] = 0] = "append";
    Strategy[Strategy["replace"] = 1] = "replace";
})(Strategy || (Strategy = {}));
export class ALuApiOptionPager {
    constructor(_service) {
        this._service = _service;
        this.outOptions$ = new Subject();
        this._loading = false;
        this._options = [];
        this._page$ = new Subject();
        this._initialized = false;
    }
    init() {
        this.initObservables();
    }
    onOpen() {
        if (!this._initialized) {
            this._page$.next(0);
            this._initialized = true;
        }
    }
    // onClose() {
    // 	this._page$.next(0);
    // }
    onScrollBottom() {
        if (!this._loading) {
            this._page$.next(this._page + 1);
        }
    }
    initObservables() {
        const _results$ = this._page$.pipe(distinctUntilChanged(), tap((p) => (this._page = p)), switchMap((page) => {
            if (page === undefined) {
                return of({ items: [], strategy: Strategy.replace });
            }
            return this._service.getPaged(page).pipe(map((items) => ({
                items: items,
                strategy: page === 0 ? Strategy.replace : Strategy.append,
            })));
        }), catchError(() => of({ items: [], strategy: Strategy.replace })), tap((results) => {
            if (results.strategy === Strategy.replace) {
                this._options = [...results.items];
            }
            else {
                this._options.push(...results.items);
            }
            this.outOptions$.next([...this._options]);
        }));
        this.loading$ = merge(this._page$.pipe(map(() => true)), _results$.pipe(map(() => false)));
        this.loading$.subscribe((l) => (this._loading = l));
    }
}
