// @lucca-front/ng/esm2022/core/date/string/string-date.adapter.mjs
import { Injectable, Inject, LOCALE_ID } from '@angular/core';
import { LuNativeDateAdapter } from '../native/index';
import { ALuDateAdapter } from '../date-adapter.class';
import * as i0 from "@angular/core";
/** bind to a string with iso 26001 format YYYY-MM-DD */
class LuStringDateAdapter extends ALuDateAdapter {
    constructor(_locale) {
        super();
        this._locale = _locale;
        this._nativeAdapter = new LuNativeDateAdapter(this._locale, {
            useUtc: true,
        });
    }
    forge(year, month, date) {
        return this.dateToString(this._nativeAdapter.forge(year, month, date));
    }
    forgeToday() {
        return this.dateToString(this._nativeAdapter.forgeToday());
    }
    forgeInvalid() {
        return 'Invalid Date';
    }
    isValid(d) {
        return this._nativeAdapter.isValid(this.stringToDate(d));
    }
    compare(a, b, granularity) {
        const da = this.stringToDate(a);
        const db = this.stringToDate(b);
        return this._nativeAdapter.compare(da, db, granularity);
    }
    isParsable(text) {
        return this._nativeAdapter.isParsable(text);
    }
    parse(text, granularity) {
        return this.dateToString(this._nativeAdapter.parse(text, granularity));
    }
    format(d, format) {
        return this._nativeAdapter.format(this.stringToDate(d), format);
    }
    clone(d) {
        return `${d}`;
    }
    getYear(d) {
        return this._nativeAdapter.getYear(this.stringToDate(d));
    }
    getMonth(d) {
        return this._nativeAdapter.getMonth(this.stringToDate(d));
    }
    getDate(d) {
        return this._nativeAdapter.getDate(this.stringToDate(d));
    }
    getDay(d) {
        return this._nativeAdapter.getDay(this.stringToDate(d));
    }
    add(d, count, granularity) {
        return this.dateToString(this._nativeAdapter.add(this.stringToDate(d), count, granularity));
    }
    stringToDate(s) {
        return new Date(`${s}T00:00:00Z`);
    }
    dateToString(d) {
        let result;
        if (d === undefined) {
            result = undefined;
        }
        else if (!this._nativeAdapter.isValid(d)) {
            result = this.forgeInvalid();
        }
        else {
            result = d.toISOString().substring(0, 10);
        }
        return result;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuStringDateAdapter, deps: [{ token: LOCALE_ID }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuStringDateAdapter }); }
}
export { LuStringDateAdapter };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuStringDateAdapter, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [LOCALE_ID]
                }] }]; } });
