// @lucca-front/ng/esm2022/formly/formly.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { LuApiModule } from '@lucca-front/ng/api';
import { LuDateModule } from '@lucca-front/ng/date';
import { LuDepartmentModule } from '@lucca-front/ng/department';
import { LuEstablishmentModule } from '@lucca-front/ng/establishment';
import { LuInputModule } from '@lucca-front/ng/input';
import { LuOptionModule } from '@lucca-front/ng/option';
import { LuQualificationModule } from '@lucca-front/ng/qualification';
import { LuSelectModule } from '@lucca-front/ng/select';
import { LuUserModule } from '@lucca-front/ng/user';
import { FormlyModule } from '@ngx-formly/core';
import { LU_FORMLY_CONFIG } from './formly.config';
import { LuFormlyFieldApi, LuFormlyFieldCheckbox, LuFormlyFieldDate, LuFormlyFieldDepartment, LuFormlyFieldEstablishment, LuFormlyFieldInput, LuFormlyFieldRadios, LuFormlyFieldSelect, LuFormlyFieldTextarea, LuFormlyFieldUser, LuFormlyFieldQualification, } from './types/index';
import { LuFormlyErrorMessage, LuFormlyWrapperCheckboxLayout, LuFormlyWrapperError, LuFormlyWrapperHelper, LuFormlyWrapperIcon, LuFormlyWrapperRadiosfieldLayout, LuFormlyWrapperSuffix, LuFormlyWrapperTextfieldLayout, } from './wrappers/index';
import * as i0 from "@angular/core";
import * as i1 from "@ngx-formly/core";
/** HACK to avoid a 'Function calls are not supported in decorators' error */
export const LuFormlyChild = FormlyModule.forChild(LU_FORMLY_CONFIG);
class LuFormlyModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyModule, declarations: [LuFormlyErrorMessage,
            LuFormlyFieldInput,
            LuFormlyFieldDate,
            LuFormlyFieldTextarea,
            LuFormlyFieldSelect,
            LuFormlyFieldUser,
            LuFormlyFieldApi,
            LuFormlyFieldDepartment,
            LuFormlyFieldEstablishment,
            LuFormlyFieldQualification,
            LuFormlyFieldRadios,
            LuFormlyFieldCheckbox,
            LuFormlyWrapperHelper,
            LuFormlyWrapperCheckboxLayout,
            LuFormlyWrapperTextfieldLayout,
            LuFormlyWrapperRadiosfieldLayout,
            LuFormlyWrapperSuffix,
            LuFormlyWrapperIcon,
            LuFormlyWrapperError], imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            LuSelectModule,
            LuOptionModule,
            LuUserModule,
            LuInputModule,
            LuApiModule,
            LuDepartmentModule,
            LuEstablishmentModule,
            LuQualificationModule,
            LuDateModule, i1.FormlyModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyModule, imports: [CommonModule,
            FormsModule,
            ReactiveFormsModule,
            LuSelectModule,
            LuOptionModule,
            LuUserModule,
            LuInputModule,
            LuApiModule,
            LuDepartmentModule,
            LuEstablishmentModule,
            LuQualificationModule,
            LuDateModule,
            LuFormlyChild] }); }
}
export { LuFormlyModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        LuFormlyErrorMessage,
                        LuFormlyFieldInput,
                        LuFormlyFieldDate,
                        LuFormlyFieldTextarea,
                        LuFormlyFieldSelect,
                        LuFormlyFieldUser,
                        LuFormlyFieldApi,
                        LuFormlyFieldDepartment,
                        LuFormlyFieldEstablishment,
                        LuFormlyFieldQualification,
                        LuFormlyFieldRadios,
                        LuFormlyFieldCheckbox,
                        LuFormlyWrapperHelper,
                        LuFormlyWrapperCheckboxLayout,
                        LuFormlyWrapperTextfieldLayout,
                        LuFormlyWrapperRadiosfieldLayout,
                        LuFormlyWrapperSuffix,
                        LuFormlyWrapperIcon,
                        LuFormlyWrapperError,
                    ],
                    imports: [
                        CommonModule,
                        FormsModule,
                        ReactiveFormsModule,
                        LuSelectModule,
                        LuOptionModule,
                        LuUserModule,
                        LuInputModule,
                        LuApiModule,
                        LuDepartmentModule,
                        LuEstablishmentModule,
                        LuQualificationModule,
                        LuDateModule,
                        LuFormlyChild,
                    ],
                }]
        }] });
