// @lucca-front/ng/esm2022/user/select/homonyms/user-homonyms.component.mjs
import { ChangeDetectionStrategy, Component, forwardRef, Inject, Optional, Self, SkipSelf } from '@angular/core';
import { ALuOptionOperator } from '@lucca-front/ng/option';
import { merge } from 'rxjs';
import { map, switchMap, withLatestFrom } from 'rxjs/operators';
import { LuUserDisplayPipe } from '../../display';
import { ALuUserHomonymsService, LuUserHomonymsService } from './user-homonyms.service';
import * as i0 from "@angular/core";
class LuUserHomonymsComponent {
    set inOptions$(in$) {
        this._outOptions$ = merge(in$, in$.pipe(map((users) => this._service.extractHomonyms(users)), switchMap((homonyms) => this._service.enrichHomonyms(homonyms)), withLatestFrom(in$), map(([enrichedUsers, users]) => {
            const result = [...users];
            enrichedUsers.forEach((user) => {
                const i = result.findIndex((u) => u.id === user.id);
                result[i] = user;
            });
            return result;
        })));
    }
    get outOptions$() {
        return this._outOptions$;
    }
    constructor(hostService, selfService) {
        this._service = hostService || selfService;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserHomonymsComponent, deps: [{ token: ALuUserHomonymsService, optional: true, skipSelf: true }, { token: ALuUserHomonymsService, self: true }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuUserHomonymsComponent, isStandalone: true, selector: "lu-user-homonyms", providers: [
            LuUserDisplayPipe,
            {
                provide: ALuOptionOperator,
                useExisting: forwardRef(() => LuUserHomonymsComponent),
                multi: true,
            },
            {
                provide: ALuUserHomonymsService,
                useClass: LuUserHomonymsService,
            },
        ], ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuUserHomonymsComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserHomonymsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'lu-user-homonyms',
                    template: '',
                    changeDetection: ChangeDetectionStrategy.OnPush,
                    standalone: true,
                    providers: [
                        LuUserDisplayPipe,
                        {
                            provide: ALuOptionOperator,
                            useExisting: forwardRef(() => LuUserHomonymsComponent),
                            multi: true,
                        },
                        {
                            provide: ALuUserHomonymsService,
                            useClass: LuUserHomonymsService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [ALuUserHomonymsService]
                }, {
                    type: Optional
                }, {
                    type: SkipSelf
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [ALuUserHomonymsService]
                }, {
                    type: Self
                }] }]; } });
