// @lucca-front/ng/esm2022/user/select/homonyms/user-homonyms.service.mjs
import { Injectable } from '@angular/core';
import { of } from 'rxjs';
import { catchError, map } from 'rxjs/operators';
import { LuDisplayFullname } from '../../display/index';
import * as i0 from "@angular/core";
import * as i1 from "../../display/index";
import * as i2 from "@angular/common/http";
export class ALuUserHomonymsService {
}
class LuUserHomonymsService extends ALuUserHomonymsService {
    extractHomonyms(users) {
        const usersByName = {};
        users.forEach((user) => {
            const name = this._pipe.transform(user, this._format);
            usersByName[name] = [...(usersByName[name] || []), user];
        });
        const nonUniqNames = Object.keys(usersByName).filter((key) => usersByName[key].length > 1);
        const homonyms = [];
        nonUniqNames.forEach((name) => homonyms.push(...usersByName[name]));
        return homonyms;
    }
    enrichHomonyms(homonyms) {
        if (!homonyms || homonyms.length === 0) {
            return of([]);
        }
        return this._http
            .get(`/api/v3/users`, {
            params: {
                id: homonyms.map((u) => u.id).join(','),
                fields: 'id,department.name',
            },
        })
            .pipe(map((res) => res.data.items), map((infos) => infos.map((info) => {
            const homonym = homonyms.find((h) => h.id === info.id);
            return {
                ...homonym,
                additionalInformation: info.department ? info.department.name : '',
            };
        })), catchError(() => of([])));
    }
    constructor(_pipe, _http) {
        super();
        this._pipe = _pipe;
        this._http = _http;
        this._format = LuDisplayFullname.lastfirst;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserHomonymsService, deps: [{ token: i1.LuUserDisplayPipe }, { token: i2.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserHomonymsService }); }
}
export { LuUserHomonymsService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserHomonymsService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.LuUserDisplayPipe }, { type: i2.HttpClient }]; } });
