// @lucca-front/ng/esm2022/date/adapter/date-adapter.pipe.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@lucca-front/ng/core";
class LuDateAdapterPipe {
    constructor(_adapter) {
        this._adapter = _adapter;
    }
    transform(d, format = 'mediumDate') {
        if (d && this._adapter.isValid(d)) {
            return this._adapter.format(d, format);
        }
        return '';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDateAdapterPipe, deps: [{ token: i1.ALuDateAdapter }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuDateAdapterPipe, isStandalone: true, name: "luDate" }); }
}
export { LuDateAdapterPipe };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDateAdapterPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'luDate',
                    pure: true,
                    standalone: true,
                }]
        }], ctorParameters: function () { return [{ type: i1.ALuDateAdapter }]; } });
