// @lucca-front/ng/esm2022/input/input.directive.mjs
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/forms";
/**
 * adds class is-filled when model is empty
 */
class LuInputDirective {
    constructor(_elementRef, _renderer, _ngControl) {
        this._elementRef = _elementRef;
        this._renderer = _renderer;
        this._ngControl = _ngControl;
    }
    isEmpty(value) {
        if (typeof value === 'string') {
            return value === '';
        }
        return value === null || value === undefined;
    }
    applyClasses(value) {
        if (this.isEmpty(value)) {
            this._renderer.removeClass(this._elementRef.nativeElement, 'is-filled');
        }
        else {
            this._renderer.addClass(this._elementRef.nativeElement, 'is-filled');
        }
    }
    ngOnInit() {
        this._ngControl.valueChanges.subscribe((v) => this.applyClasses(v));
        const val = this._ngControl.value;
        this.applyClasses(val);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuInputDirective, deps: [{ token: i0.ElementRef }, { token: i0.Renderer2 }, { token: i1.NgControl }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuInputDirective, isStandalone: true, selector: "[luInput]", ngImport: i0 }); }
}
export { LuInputDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuInputDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luInput]',
                    standalone: true,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }, { type: i0.Renderer2 }, { type: i1.NgControl }]; } });
