// @lucca-front/ng/esm2022/input/input.model.mjs
export class ALuInput {
    get placeholder() {
        return this._placeholder;
    }
    constructor(_changeDetectorRef, _elementRef, _renderer) {
        this._changeDetectorRef = _changeDetectorRef;
        this._elementRef = _elementRef;
        this._renderer = _renderer;
        // From ControlValueAccessor interface
        this._cvaOnChange = () => {
            return;
        };
        // From ControlValueAccessor interface
        this._onTouched = () => {
            return;
        };
    }
    setValue(value) {
        this.value = value;
        this._cvaOnChange(value);
        this._onTouched();
    }
    get value() {
        return this._value;
    }
    set value(value) {
        this._value = value;
        this.render();
        this.applyClasses();
        this._changeDetectorRef.markForCheck();
    }
    // From ControlValueAccessor interface
    writeValue(value) {
        this.value = value;
    }
    registerOnChange(fn) {
        this._cvaOnChange = fn;
    }
    registerOnTouched(fn) {
        this._onTouched = fn;
    }
    isEmpty() {
        return this.value === null || this.value === undefined;
    }
    applyClasses() {
        if (this.isEmpty()) {
            this._renderer.removeClass(this._elementRef.nativeElement, 'is-filled');
        }
        else {
            this._renderer.addClass(this._elementRef.nativeElement, 'is-filled');
        }
    }
}
