// @lucca-front/ng/esm2022/formly/wrappers/error.mjs
import { ChangeDetectionStrategy, Component, Input, ViewChild, ViewContainerRef } from '@angular/core';
import { FieldWrapper } from '@ngx-formly/core';
import { buildAddWrapperExtension } from '../formly.utils';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/forms";
// wrapper
class LuFormlyWrapperError extends FieldWrapper {
    get validationId() {
        return this.field.id + '-message';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyWrapperError, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuFormlyWrapperError, selector: "lu-formly-wrapper-error", viewQueries: [{ propertyName: "fieldComponent", first: true, predicate: ["fieldComponent"], descendants: true, read: ViewContainerRef, static: true }], usesInheritance: true, ngImport: i0, template: "<ng-container #fieldComponent></ng-container>\n<div class=\"textfield-messages\" *ngIf=\"formControl.touched && formControl.invalid\" [id]=\"validationId\">\n\t<lu-formly-error-message [formControl]=\"formControl\" [field]=\"field\"></lu-formly-error-message>\n</div>\n", styles: [":host{display:inherit;flex-direction:inherit;flex:1 1 auto;width:100%}\n"], dependencies: [{ kind: "directive", type: i0.forwardRef(function () { return i1.NgIf; }), selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i0.forwardRef(function () { return i2.NgControlStatus; }), selector: "[formControlName],[ngModel],[formControl]" }, { kind: "directive", type: i0.forwardRef(function () { return i2.FormControlDirective; }), selector: "[formControl]", inputs: ["formControl", "disabled", "ngModel"], outputs: ["ngModelChange"], exportAs: ["ngForm"] }, { kind: "component", type: i0.forwardRef(function () { return LuFormlyErrorMessage; }), selector: "lu-formly-error-message", inputs: ["formControl", "field"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuFormlyWrapperError };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyWrapperError, decorators: [{
            type: Component,
            args: [{ selector: 'lu-formly-wrapper-error', changeDetection: ChangeDetectionStrategy.OnPush, template: "<ng-container #fieldComponent></ng-container>\n<div class=\"textfield-messages\" *ngIf=\"formControl.touched && formControl.invalid\" [id]=\"validationId\">\n\t<lu-formly-error-message [formControl]=\"formControl\" [field]=\"field\"></lu-formly-error-message>\n</div>\n", styles: [":host{display:inherit;flex-direction:inherit;flex:1 1 auto;width:100%}\n"] }]
        }], propDecorators: { fieldComponent: [{
                type: ViewChild,
                args: ['fieldComponent', { read: ViewContainerRef, static: true }]
            }] } });
// component that display the right error message
class LuFormlyErrorMessage {
    get errorMessages() {
        const messages = [];
        if (this.formControl?.errors) {
            Object.keys(this.formControl.errors).forEach((key) => {
                if (this.field.validation?.messages?.[key]) {
                    messages.push(this.field.validation.messages[key]);
                }
            });
        }
        return messages;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyErrorMessage, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuFormlyErrorMessage, selector: "lu-formly-error-message", inputs: { formControl: "formControl", field: "field" }, ngImport: i0, template: `<div class="textfield-messages-error" *ngFor="let message of errorMessages">
		{{ message }}
	</div>`, isInline: true, dependencies: [{ kind: "directive", type: i1.NgForOf, selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuFormlyErrorMessage };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyErrorMessage, decorators: [{
            type: Component,
            args: [{
                    selector: 'lu-formly-error-message',
                    template: `<div class="textfield-messages-error" *ngFor="let message of errorMessages">
		{{ message }}
	</div>`,
                    changeDetection: ChangeDetectionStrategy.OnPush,
                }]
        }], propDecorators: { formControl: [{
                type: Input
            }], field: [{
                type: Input
            }] } });
export const templateErrorExtension = buildAddWrapperExtension('error', (field) => !!field.validation);
