// @lucca-front/ng/esm2022/formly/wrappers/radiosfield-layout.mjs
import { ChangeDetectionStrategy, Component, ViewChild, ViewContainerRef } from '@angular/core';
import { FieldWrapper } from '@ngx-formly/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
// wrapper component
class LuFormlyWrapperRadiosfieldLayout extends FieldWrapper {
    get mod() {
        return (this.props['mod'] || '');
    }
    get isRequired() {
        return !!this.props && !!this.props.required ? 'is-required' : '';
    }
    get isFocused() {
        return !!this.props && this.props['_isFocused'] ? 'is-focused' : '';
    }
    get isError() {
        return this.formControl.invalid && this.formControl.touched ? 'is-error' : '';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyWrapperRadiosfieldLayout, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuFormlyWrapperRadiosfieldLayout, selector: "lu-formly-wrapper-radiosfield-layout", viewQueries: [{ propertyName: "fieldComponent", first: true, predicate: ["fieldComponent"], descendants: true, read: ViewContainerRef, static: true }], usesInheritance: true, ngImport: i0, template: "<div class=\"fieldsetWrapper\">\n\t<fieldset class=\"radiosfield\" [ngClass]=\"[mod, isRequired, isFocused, isError]\">\n\t\t<ng-container #fieldComponent></ng-container>\n\t</fieldset>\n</div>\n", styles: [":host{display:inherit;flex-direction:inherit;flex:1 1 auto;width:100%}\n"], dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuFormlyWrapperRadiosfieldLayout };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyWrapperRadiosfieldLayout, decorators: [{
            type: Component,
            args: [{ selector: 'lu-formly-wrapper-radiosfield-layout', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div class=\"fieldsetWrapper\">\n\t<fieldset class=\"radiosfield\" [ngClass]=\"[mod, isRequired, isFocused, isError]\">\n\t\t<ng-container #fieldComponent></ng-container>\n\t</fieldset>\n</div>\n", styles: [":host{display:inherit;flex-direction:inherit;flex:1 1 auto;width:100%}\n"] }]
        }], propDecorators: { fieldComponent: [{
                type: ViewChild,
                args: ['fieldComponent', { read: ViewContainerRef, static: true }]
            }] } });
