// @lucca-front/ng/esm2022/formly/wrappers/textfield-layout.mjs
import { ChangeDetectionStrategy, Component, ViewChild, ViewContainerRef } from '@angular/core';
import { FieldWrapper } from '@ngx-formly/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
// wrapper component
class LuFormlyWrapperTextfieldLayout extends FieldWrapper {
    get mod() {
        return (this.props['mod'] || '');
    }
    get modMultiline() {
        return !!this.field && this.field.type === 'textarea' ? 'mod-multiline' : '';
    }
    get modWithSuffix() {
        return !!this.props && !!this.props['suffix'] ? 'mod-withSuffix' : '';
    }
    get isRequired() {
        return !!this.props && !!this.props.required ? 'is-required' : '';
    }
    get isDisabled() {
        return !!this.props && !!this.props.disabled ? 'is-disabled' : '';
    }
    get isFocused() {
        return !!this.props && this.props['_isFocused'] ? 'is-focused' : '';
    }
    get isError() {
        return this.formControl.invalid && this.formControl.touched ? 'is-error' : '';
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyWrapperTextfieldLayout, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuFormlyWrapperTextfieldLayout, selector: "lu-formly-wrapper-layout", viewQueries: [{ propertyName: "fieldComponent", first: true, predicate: ["fieldComponent"], descendants: true, read: ViewContainerRef, static: true }], usesInheritance: true, ngImport: i0, template: "<label class=\"textfield\" [ngClass]=\"[mod, modWithSuffix, modMultiline, isRequired, isFocused, isError, isDisabled]\">\n\t<ng-container #fieldComponent></ng-container>\n</label>\n", styles: [":host{display:inherit;flex-direction:inherit;flex:1 1 auto;width:100%}\n"], dependencies: [{ kind: "directive", type: i1.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuFormlyWrapperTextfieldLayout };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuFormlyWrapperTextfieldLayout, decorators: [{
            type: Component,
            args: [{ selector: 'lu-formly-wrapper-layout', changeDetection: ChangeDetectionStrategy.OnPush, template: "<label class=\"textfield\" [ngClass]=\"[mod, modWithSuffix, modMultiline, isRequired, isFocused, isError, isDisabled]\">\n\t<ng-container #fieldComponent></ng-container>\n</label>\n", styles: [":host{display:inherit;flex-direction:inherit;flex:1 1 auto;width:100%}\n"] }]
        }], propDecorators: { fieldComponent: [{
                type: ViewChild,
                args: ['fieldComponent', { read: ViewContainerRef, static: true }]
            }] } });
