// @lucca-front/ng/esm2022/option/operator/for-groups/for-groups.directive.mjs
import { Directive, forwardRef, Input } from '@angular/core';
import { Subscription } from 'rxjs';
import { ALuOptionOperator } from '../option-operator.model';
import * as i0 from "@angular/core";
export class LuForGroupContext {
    constructor($implicit, index, count) {
        this.$implicit = $implicit;
        this.index = index;
        this.count = count;
    }
    get first() {
        return this.index === 0;
    }
    get last() {
        return this.index === this.count - 1;
    }
    get even() {
        return this.index % 2 === 0;
    }
    get odd() {
        return !this.even;
    }
}
class LuForGroupsDirective {
    set attrGroupBy(fn) {
        this._groupByFn = fn;
    }
    set inOptions$(options$) {
        this._subs.add(options$.subscribe((options) => this.render(options)));
        this.outOptions$ = options$;
    }
    constructor(_vcr, _cdr, _templateRef) {
        this._vcr = _vcr;
        this._cdr = _cdr;
        this._templateRef = _templateRef;
        this._subs = new Subscription();
    }
    ngOnDestroy() {
        this._subs.unsubscribe();
    }
    render(options) {
        this._vcr.clear();
        const count = options.length;
        const groups = this.groupBy(options);
        groups.forEach((group, index) => {
            this._vcr.createEmbeddedView(this._templateRef, new LuForGroupContext(group, index, count));
        });
        this._cdr.markForCheck();
    }
    groupBy(items) {
        const groups = [];
        items.forEach((item) => {
            const key = this._groupByFn(item);
            let group = groups.find((g) => g.key === key);
            if (!group) {
                group = { key: key, items: [] };
                groups.push(group);
            }
            group.items.push(item);
        });
        return groups;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForGroupsDirective, deps: [{ token: i0.ViewContainerRef }, { token: i0.ChangeDetectorRef }, { token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuForGroupsDirective, isStandalone: true, selector: "[luForGroups]", inputs: { attrGroupBy: ["luForGroupsGroupBy", "attrGroupBy"] }, providers: [
            {
                provide: ALuOptionOperator,
                useExisting: forwardRef(() => LuForGroupsDirective),
                multi: true,
            },
        ], ngImport: i0 }); }
}
export { LuForGroupsDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForGroupsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luForGroups]',
                    standalone: true,
                    providers: [
                        {
                            provide: ALuOptionOperator,
                            useExisting: forwardRef(() => LuForGroupsDirective),
                            multi: true,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.ChangeDetectorRef }, { type: i0.TemplateRef }]; }, propDecorators: { attrGroupBy: [{
                type: Input,
                args: ['luForGroupsGroupBy']
            }] } });
