// @lucca-front/ng/esm2022/date/calendar/calendar-item.factory.mjs
import { Injectable } from '@angular/core';
import { DayItem, DecadeItem, MonthItem, YearItem } from './calendar-item.class';
import * as i0 from "@angular/core";
import * as i1 from "@lucca-front/ng/core";
class LuCalendarItemFactory {
    constructor(_adapter) {
        this._adapter = _adapter;
    }
    forgeDay(d, format = 'd') {
        const date = this._adapter.clone(d);
        return new DayItem(date, this._adapter.format(date, format));
    }
    forgeMonth(d, format = 'MMM') {
        const year = this._adapter.getYear(d);
        const month = this._adapter.getMonth(d);
        const monthStart = this._adapter.forge(year, month, 1);
        return new MonthItem(monthStart, this._adapter.format(monthStart, format));
    }
    forgeYear(d, format = 'y') {
        const year = this._adapter.getYear(d);
        const yearStart = this._adapter.forge(year, 1, 1);
        return new YearItem(yearStart, this._adapter.format(yearStart, format));
    }
    forgeDecade(d, format = 'y') {
        const year = this._adapter.getYear(d);
        const decadeStart = this._adapter.forge(10 * Math.floor(year / 10), 1, 1);
        const decadeEnd = this._adapter.forge(10 * Math.floor(year / 10) + 9, 1, 1);
        const label = `${this._adapter.format(decadeStart, format)} - ${this._adapter.format(decadeEnd, format)}`;
        return new DecadeItem(decadeStart, label);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuCalendarItemFactory, deps: [{ token: i1.ALuDateAdapter }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuCalendarItemFactory }); }
}
export { LuCalendarItemFactory };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuCalendarItemFactory, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.ALuDateAdapter }]; } });
