// @lucca-front/ng/esm2022/date/calendar/calendar-item.class.mjs
import { ELuDateGranularity } from '@lucca-front/ng/core';
export class ACalendarItem {
    constructor() {
        this.mods = [];
        this.isDisabled = false;
    }
    get id() {
        return `${this.granularity}-${this.date.toString()}`;
    }
}
export class DayItem extends ACalendarItem {
    constructor(date, label) {
        super();
        this.date = date;
        this.label = label;
        this.granularity = ELuDateGranularity.day;
    }
}
export class MonthItem extends ACalendarItem {
    constructor(date, label) {
        super();
        this.date = date;
        this.label = label;
        this.granularity = ELuDateGranularity.month;
    }
}
export class YearItem extends ACalendarItem {
    constructor(date, label) {
        super();
        this.date = date;
        this.label = label;
        this.granularity = ELuDateGranularity.year;
    }
}
export class DecadeItem extends ACalendarItem {
    constructor(date, label) {
        super();
        this.date = date;
        this.label = label;
        this.granularity = ELuDateGranularity.decade;
    }
}
