// @lucca-front/ng/esm2022/user/tile/user-tile.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LuUserDisplayModule } from '../display/index';
import { LuUserPictureModule } from '../picture/index';
import { LuUserTileComponent } from './user-tile.component';
import * as i0 from "@angular/core";
class LuUserTileModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserTileModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuUserTileModule, declarations: [LuUserTileComponent], imports: [CommonModule, LuUserPictureModule, LuUserDisplayModule], exports: [LuUserTileComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserTileModule, imports: [CommonModule, LuUserPictureModule, LuUserDisplayModule] }); }
}
export { LuUserTileModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserTileModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LuUserPictureModule, LuUserDisplayModule],
                    declarations: [LuUserTileComponent],
                    exports: [LuUserTileComponent],
                }]
        }] });
