// @lucca-front/ng/esm2022/user/service/user-v3.service.mjs
import { Injectable } from '@angular/core';
import { LuApiV3Service } from '@lucca-front/ng/api';
import { map, switchMap } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
class LuUserV3Service extends LuApiV3Service {
    set appInstanceId(appInstanceId) {
        this._appInstanceId = appInstanceId;
    }
    set operations(operations) {
        this._operations = operations;
    }
    constructor(_http) {
        super(_http);
        this._http = _http;
        this._operations = [];
        this.api = '/api/v3/users/search';
        this.fields = 'id,firstname,lastname,picture[href]';
        this.orderBy = 'lastname,asc,firstname,asc';
    }
    getMe() {
        return this._http.get(`/api/v3/users/me?fields=id`).pipe(switchMap((r) => this._get(this.url + `&id=${r.data.id}`)), map((users) => users[0]));
    }
    get url() {
        if (!this._appInstanceId || !this._operations || !this._operations.length) {
            return `${this._api}?${[...this._filters, this._orderBy, this._fields].filter((f) => !!f).join('&')}`;
        }
        else {
            return `/api/v3/users/scopedsearch?${[...this._filters, `appInstanceId=${this._appInstanceId}`, `operations=${this._operations.join(',')}`, this._orderBy, this._fields]
                .filter((f) => !!f)
                .join('&')}`;
        }
    }
    // FIXME typing
    _get(url) {
        return super._get(url).pipe(map((suggestions) => suggestions.map((s) => s.item)));
    }
    _clueFilter(clue) {
        const urlSafeClue = clue
            .split(' ')
            .map((c) => encodeURIComponent(c))
            .join(',');
        return `clue=${urlSafeClue}`;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserV3Service, deps: [{ token: i1.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserV3Service }); }
}
export { LuUserV3Service };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserV3Service, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.HttpClient }]; } });
