// @lucca-front/ng/esm2022/establishment/select/for-legal-units/for-legal-units.directive.mjs
import { Directive, forwardRef } from '@angular/core';
import { ALuOptionOperator, LuForGroupsDirective } from '@lucca-front/ng/option';
import * as i0 from "@angular/core";
class LuForLegalUnitsDirective extends LuForGroupsDirective {
    constructor(_vcr, _cdr, _templateRef) {
        super(_vcr, _cdr, _templateRef);
        this._vcr = _vcr;
        this._cdr = _cdr;
        this._templateRef = _templateRef;
    }
    groupBy(items) {
        const groups = [];
        items.forEach((item) => {
            const legalUnit = item?.legalUnit;
            let group = groups.find((g) => g.key.id === legalUnit.id);
            if (!group) {
                group = { key: legalUnit, items: [] };
                groups.push(group);
            }
            group.items.push(item);
        });
        return groups;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForLegalUnitsDirective, deps: [{ token: i0.ViewContainerRef }, { token: i0.ChangeDetectorRef }, { token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuForLegalUnitsDirective, isStandalone: true, selector: "[luForLegalUnits]", providers: [
            {
                provide: ALuOptionOperator,
                useExisting: forwardRef(() => LuForLegalUnitsDirective),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0 }); }
}
export { LuForLegalUnitsDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuForLegalUnitsDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luForLegalUnits]',
                    standalone: true,
                    providers: [
                        {
                            provide: ALuOptionOperator,
                            useExisting: forwardRef(() => LuForLegalUnitsDirective),
                            multi: true,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i0.ViewContainerRef }, { type: i0.ChangeDetectorRef }, { type: i0.TemplateRef }]; } });
