// @lucca-front/ng/esm2022/user/picture/user-picture.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { LuUserDisplayModule } from '../display/index';
import { LuUserPictureComponent } from './user-picture.component';
import * as i0 from "@angular/core";
class LuUserPictureModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserPictureModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuUserPictureModule, declarations: [LuUserPictureComponent], imports: [CommonModule, LuUserDisplayModule], exports: [LuUserPictureComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserPictureModule, imports: [CommonModule, LuUserDisplayModule] }); }
}
export { LuUserPictureModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuUserPictureModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule, LuUserDisplayModule],
                    declarations: [LuUserPictureComponent],
                    exports: [LuUserPictureComponent],
                }]
        }] });
