// @lucca-front/ng/esm2022/establishment/select/legal-unit-selector/legal-unit-selector.directive.mjs
import { Directive, forwardRef, HostListener, Inject, Input, Optional, Self, SkipSelf } from '@angular/core';
import { ALuOptionSelector } from '@lucca-front/ng/option';
import { Subject, Subscription } from 'rxjs';
import { ALuEstablishmentService, LuEstablishmentService } from '../../service/index';
import * as i0 from "@angular/core";
import * as i1 from "../../service/index";
class LuLegalUnitSelectorDirective {
    set filters(filters) {
        this._service.filters = filters;
    }
    set appInstanceId(appId) {
        this._service.appInstanceId = appId;
    }
    set operations(ops) {
        this._service.operations = ops;
    }
    constructor(hostService, selfService) {
        this.multiple = true;
        this.onSelectValue = new Subject();
        this._subs = new Subscription();
        this._service = hostService || selfService;
    }
    onClick() {
        const sub = this._service.getAll([`legalUnitId=${this.legalUnit.id}`]).subscribe((establishments) => {
            if (this.shouldAdd(establishments)) {
                const selectedEstablishmentIds = new Set((this._values ?? []).map((ets) => ets.id));
                this.onSelectValue.next(Array.from([...(this._values ?? []), ...establishments.filter((ets) => !selectedEstablishmentIds.has(ets.id))]));
            }
            else {
                const establishmentIds = new Set(establishments.map((ets) => ets.id));
                this.onSelectValue.next((this._values ?? []).filter((ets) => !establishmentIds.has(ets.id)));
            }
        });
        this._subs.add(sub);
    }
    setValue(values) {
        this._values = values;
    }
    ngOnDestroy() {
        this._subs.unsubscribe();
    }
    shouldAdd(establishments) {
        const selectedCount = (this._values ?? []).filter((ets) => ets.legalUnitId === this.legalUnit.id).length;
        return establishments.length > selectedCount;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuLegalUnitSelectorDirective, deps: [{ token: ALuEstablishmentService, optional: true, skipSelf: true }, { token: ALuEstablishmentService, self: true }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuLegalUnitSelectorDirective, isStandalone: true, selector: "[luLegalUnitSelector]", inputs: { legalUnit: ["luLegalUnitSelector", "legalUnit"], filters: ["luLegalUnitSelectorFilters", "filters"], appInstanceId: ["luLegalUnitSelectorAppInstanceId", "appInstanceId"], operations: ["luLegalUnitSelectorOperations", "operations"] }, host: { listeners: { "click": "onClick()" } }, providers: [
            {
                provide: ALuOptionSelector,
                useExisting: forwardRef(() => LuLegalUnitSelectorDirective),
                multi: true,
            },
            {
                provide: ALuEstablishmentService,
                useClass: LuEstablishmentService,
            },
        ], ngImport: i0 }); }
}
export { LuLegalUnitSelectorDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuLegalUnitSelectorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luLegalUnitSelector]',
                    standalone: true,
                    providers: [
                        {
                            provide: ALuOptionSelector,
                            useExisting: forwardRef(() => LuLegalUnitSelectorDirective),
                            multi: true,
                        },
                        {
                            provide: ALuEstablishmentService,
                            useClass: LuEstablishmentService,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i1.LuEstablishmentService, decorators: [{
                    type: Inject,
                    args: [ALuEstablishmentService]
                }, {
                    type: Optional
                }, {
                    type: SkipSelf
                }] }, { type: i1.LuEstablishmentService, decorators: [{
                    type: Inject,
                    args: [ALuEstablishmentService]
                }, {
                    type: Self
                }] }]; }, propDecorators: { legalUnit: [{
                type: Input,
                args: ['luLegalUnitSelector']
            }], filters: [{
                type: Input,
                args: ['luLegalUnitSelectorFilters']
            }], appInstanceId: [{
                type: Input,
                args: ['luLegalUnitSelectorAppInstanceId']
            }], operations: [{
                type: Input,
                args: ['luLegalUnitSelectorOperations']
            }], onClick: [{
                type: HostListener,
                args: ['click']
            }] } });
