// @lucca-front/ng/esm2022/date/input/date-input.translate.mjs
import { InjectionToken } from '@angular/core';
export const LU_DATE_INPUT_TRANSLATIONS = new InjectionToken('LuDateInputtranslations', {
    factory: () => luDateInputTranslations,
});
export const luDateInputTranslations = {
    en: {
        placeholderDay: 'DD/MM/YYYY',
        placeholderMonth: 'MM/YYYY',
        placeholderYear: 'YYYY',
        formatDay: 'shortDate',
        formatMonth: 'MM/Y',
        formatYear: 'Y',
    },
    'en-GB': {
        placeholderDay: 'DD/MM/YYYY',
        placeholderMonth: 'MM/YYYY',
        placeholderYear: 'YYYY',
        formatDay: 'shortDate',
        formatMonth: 'MM/Y',
        formatYear: 'Y',
    },
    'en-US': {
        placeholderDay: 'MM/DD/YYYY',
        placeholderMonth: 'MM/YYYY',
        placeholderYear: 'YYYY',
        formatDay: 'shortDate',
        formatMonth: 'MM/Y',
        formatYear: 'Y',
    },
    fr: {
        placeholderDay: 'JJ/MM/AAAA',
        placeholderMonth: 'MM/AAAA',
        placeholderYear: 'AAAA',
        formatDay: 'shortDate',
        formatMonth: 'MM/Y',
        formatYear: 'Y',
    },
};
