// @lucca-front/ng/esm2022/core-select/input/select-input.models.mjs
import { Overlay, OverlayContainer } from '@angular/cdk/overlay';
import { Platform } from '@angular/cdk/platform';
import { DOCUMENT } from '@angular/common';
import { ElementRef, Injectable, inject } from '@angular/core';
import { SELECT_ID, SELECT_LABEL, SELECT_LABEL_ID } from '../select.model';
import * as i0 from "@angular/core";
let selectId = 0;
function selectIdFactory() {
    return selectId++;
}
function selectLabelFactory() {
    const elementRef = inject(ElementRef);
    function getLabel(node) {
        if (node instanceof HTMLLabelElement) {
            return node;
        }
        if (!node.parentElement) {
            return undefined;
        }
        return getLabel(node.parentElement);
    }
    return getLabel(elementRef.nativeElement);
}
function selectLabelIdFactory() {
    return inject(SELECT_LABEL)?.id || `lu-select-label-${inject(SELECT_ID)}`;
}
export function provideLuSelectLabelsAndIds() {
    return [
        { provide: SELECT_ID, useFactory: selectIdFactory },
        { provide: SELECT_LABEL, useFactory: selectLabelFactory },
        { provide: SELECT_LABEL_ID, useFactory: selectLabelIdFactory },
    ];
}
class LuSelectOverlayContainer extends OverlayContainer {
    constructor() {
        super(inject(DOCUMENT), inject(Platform));
        this.selectLabelId = inject(SELECT_LABEL_ID);
        this.selectId = inject(SELECT_ID);
    }
    _createContainer() {
        super._createContainer();
        this._containerElement.setAttribute('aria-labelledby', this.selectLabelId);
        this._containerElement.id = `lu-select-overlay-container-${this.selectId}`;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSelectOverlayContainer, deps: [], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSelectOverlayContainer }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSelectOverlayContainer, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return []; } });
export function provideLuSelectOverlayContainer() {
    return [
        Overlay,
        {
            provide: OverlayContainer,
            useClass: LuSelectOverlayContainer,
        },
    ];
}
