// @lucca-front/ng/esm2022/dropdown/trigger/dropdown-trigger.module.mjs
import { NgModule } from '@angular/core';
import { LuDropdownTriggerDirective } from './dropdown-trigger.directive';
import { OverlayModule } from '@angular/cdk/overlay';
import * as i0 from "@angular/core";
class LuDropdownTriggerModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownTriggerModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownTriggerModule, declarations: [LuDropdownTriggerDirective], imports: [OverlayModule], exports: [LuDropdownTriggerDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownTriggerModule, imports: [OverlayModule] }); }
}
export { LuDropdownTriggerModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownTriggerModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LuDropdownTriggerDirective],
                    imports: [OverlayModule],
                    exports: [LuDropdownTriggerDirective],
                }]
        }] });
