// @lucca-front/ng/esm2022/department/select/feeder/department-feeder.component.mjs
import { ChangeDetectionStrategy, Component, forwardRef, Inject, Input, Optional, Self, SkipSelf } from '@angular/core';
import { ALuOnOpenSubscriber } from '@lucca-front/ng/core';
import { ALuTreeOptionOperator } from '@lucca-front/ng/option';
import { Subject } from 'rxjs';
import { ALuDepartmentService, LuDepartmentV3Service } from '../../service/index';
import * as i0 from "@angular/core";
import * as i1 from "../../service/index";
class LuDepartmentFeederComponent extends ALuTreeOptionOperator {
    set appInstanceId(appInstanceId) {
        this._service.appInstanceId = appInstanceId;
    }
    set operations(operations) {
        this._service.operations = operations;
    }
    set filters(filters) {
        this._service.filters = filters;
    }
    constructor(hostService, selfService) {
        super();
        this._out$ = new Subject();
        this._service = hostService || selfService;
        this.outOptions$ = this._out$.asObservable();
    }
    onOpen() {
        this._service.getTrees().subscribe((trees) => this._out$.next(trees));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDepartmentFeederComponent, deps: [{ token: ALuDepartmentService, optional: true, skipSelf: true }, { token: ALuDepartmentService, self: true }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuDepartmentFeederComponent, isStandalone: true, selector: "lu-department-feeder", inputs: { appInstanceId: "appInstanceId", operations: "operations", filters: "filters" }, providers: [
            {
                provide: ALuTreeOptionOperator,
                useExisting: forwardRef(() => LuDepartmentFeederComponent),
                multi: true,
            },
            {
                provide: ALuDepartmentService,
                useClass: LuDepartmentV3Service,
            },
            {
                provide: ALuOnOpenSubscriber,
                useExisting: forwardRef(() => LuDepartmentFeederComponent),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuDepartmentFeederComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDepartmentFeederComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lu-department-feeder', template: '', changeDetection: ChangeDetectionStrategy.OnPush, standalone: true, providers: [
                        {
                            provide: ALuTreeOptionOperator,
                            useExisting: forwardRef(() => LuDepartmentFeederComponent),
                            multi: true,
                        },
                        {
                            provide: ALuDepartmentService,
                            useClass: LuDepartmentV3Service,
                        },
                        {
                            provide: ALuOnOpenSubscriber,
                            useExisting: forwardRef(() => LuDepartmentFeederComponent),
                            multi: true,
                        },
                    ] }]
        }], ctorParameters: function () { return [{ type: i1.LuDepartmentV3Service, decorators: [{
                    type: Inject,
                    args: [ALuDepartmentService]
                }, {
                    type: Optional
                }, {
                    type: SkipSelf
                }] }, { type: i1.LuDepartmentV3Service, decorators: [{
                    type: Inject,
                    args: [ALuDepartmentService]
                }, {
                    type: Self
                }] }]; }, propDecorators: { appInstanceId: [{
                type: Input
            }], operations: [{
                type: Input
            }], filters: [{
                type: Input
            }] } });
