// @lucca-front/ng/esm2022/popover/target/popover-target.model.mjs
export class ALuPopoverTarget {
    constructor() {
        this._position = 'below';
        this._alignment = 'center';
        this._overlap = false;
        this._offsetX = 0;
        this._offsetY = 0;
    }
    /** the element used to position the panel */
    get elementRef() {
        return this._elementRef;
    }
    set elementRef(ref) {
        this._elementRef = ref;
    }
    /** how you want to position the panel relative to the target, allowed values: above, below, before, after */
    get position() {
        return this._position;
    }
    set position(position) {
        if (position !== 'above' && position !== 'below' && position !== 'after' && position !== 'before') {
            throw Error(`LuPopoverPosition value must be 'above', 'below', 'before' or 'after'. Got "${position}".
      Example: <lu-popover [position]="'before'" #popover="LuPopover"></lu-popover>`);
        }
        this._position = position;
        // this.setPositionClasses(this._position, this._alignment);
    }
    /** how the panel will be align with the target, allowed values: top, bottom, left, right */
    get alignment() {
        return this._alignment;
    }
    set alignment(alignment) {
        if (alignment !== 'center' && alignment !== 'top' && alignment !== 'bottom' && alignment !== 'right' && alignment !== 'left') {
            throw Error(`LuPopoverAlignment value must be 'top', 'bottom', 'right' or 'left'. Got "${alignment}".
      Example: <lu-popover [position]="above" [alignment]="left" #popover="LuPopover"></lu-popover>`);
        }
        this._alignment = alignment;
        // this.setPositionClasses(this._position, this._alignment);
    }
    /** set to true if you want the panel to appear on top of the target */
    get overlap() {
        return this._overlap;
    }
    set overlap(ot) {
        this._overlap = ot;
    }
    get offsetX() {
        return this._offsetX;
    }
    set offsetX(ox) {
        this._offsetX = ox;
    }
    get offsetY() {
        return this._offsetY;
    }
    set offsetY(oy) {
        this._offsetY = oy;
    }
}
export class LuPopoverTarget extends ALuPopoverTarget {
    constructor() {
        super();
    }
}
