// @lucca-front/ng/esm2022/modal/modal-ref.factory.mjs
import { ComponentPortal } from '@angular/cdk/portal';
import { ApplicationRef, ChangeDetectionStrategy, inject, Injectable, Injector } from '@angular/core';
import { LuModalPanelComponent, LuModalPanelComponentDefaultCD } from './modal-panel.component';
import { ALuModalRef } from './modal-ref.model';
import { LU_MODAL_DATA } from './modal.token';
import { setAriaHiddenOnApplicationRoot } from './modal.utils';
import * as i0 from "@angular/core";
import * as i1 from "@angular/cdk/overlay";
class LuModalRef extends ALuModalRef {
    constructor(_overlay, _injector, _component, _config, _applicationRef) {
        super(_overlay, _injector, _component, _config);
        this._overlay = _overlay;
        this._injector = _injector;
        this._component = _component;
        this._config = _config;
        this._applicationRef = _applicationRef;
    }
    _openPopup(data) {
        const injector = Injector.create({
            providers: [
                { provide: ALuModalRef, useValue: this },
                { provide: LU_MODAL_DATA, useValue: data },
            ],
            parent: this._injector,
        });
        const containerPortal = new ComponentPortal(this._config.changeDetection === ChangeDetectionStrategy.OnPush ? LuModalPanelComponent : LuModalPanelComponentDefaultCD, undefined, injector);
        this._containerRef = this._overlayRef.attach(containerPortal);
        const panel = this._containerRef.instance;
        this._componentRef = panel.attachInnerComponent(this._component, injector);
        setAriaHiddenOnApplicationRoot(this._applicationRef, true);
    }
    _closePopup() {
        this._componentRef.destroy();
        this._containerRef.destroy();
        setAriaHiddenOnApplicationRoot(this._applicationRef, false);
    }
    _getOverlayPanelClasses() {
        const classes = super._getOverlayPanelClasses();
        if (this.modalClasses.overlayPane) {
            classes.push(this.modalClasses.overlayPane);
        }
        return classes;
    }
}
class LuModalRefFactory {
    constructor(_overlay, _injector) {
        this._overlay = _overlay;
        this._injector = _injector;
        this._applicationRef = inject(ApplicationRef);
    }
    forge(component, config) {
        return new LuModalRef(this._overlay, this._injector, component, config, this._applicationRef);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuModalRefFactory, deps: [{ token: i1.Overlay }, { token: i0.Injector }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuModalRefFactory }); }
}
export { LuModalRefFactory };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuModalRefFactory, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Overlay }, { type: i0.Injector }]; } });
