// @lucca-front/ng/esm2022/modal/modal.module.mjs
import { OverlayModule } from '@angular/cdk/overlay';
import { NgModule } from '@angular/core';
import { LuModalRefFactory } from './modal-ref.factory';
import { LuModal } from './modal.service';
import { LU_MODAL_REF_FACTORY } from './modal.token';
import * as i0 from "@angular/core";
/**
 * Provide LuModal.
 * Note that OverlayModule should be imported in one of the EnvironmentInjectors (AppModule, lazy-loaded route) using `providers: [importProvidersFrom(OverlayModule)]`.
 */
export function provideLuModal() {
    return [LuModal, { provide: LU_MODAL_REF_FACTORY, useClass: LuModalRefFactory }];
}
class LuModalModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuModalModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuModalModule, imports: [OverlayModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuModalModule, providers: [provideLuModal()], imports: [OverlayModule] }); }
}
export { LuModalModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuModalModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [OverlayModule],
                    providers: [provideLuModal()],
                }]
        }] });
