// @lucca-front/ng/esm2022/simple-select/api/api-v4.directive.mjs
import { HttpClient } from '@angular/common/http';
import { Directive, inject, Input } from '@angular/core';
import { BehaviorSubject, combineLatest, map, ReplaySubject, switchMap, take } from 'rxjs';
import { ALuSimpleSelectApiDirective } from './api.directive';
import * as i0 from "@angular/core";
class LuSimpleSelectApiV4Directive extends ALuSimpleSelectApiDirective {
    constructor() {
        super(...arguments);
        this.url$ = new ReplaySubject(1);
        this.sort$ = new BehaviorSubject('+name');
        this.filters$ = new BehaviorSubject({});
        this.httpClient = inject(HttpClient);
        this.params$ = combineLatest([this.filters$, this.sort$, this.clue$]).pipe(map(([filters, sort, clue]) => ({
            ...filters,
            ...(sort ? { sort } : {}),
            ...(clue ? { search: clue } : {}),
        })));
        this.optionComparer = (a, b) => a.id === b.id;
    }
    set apiV4(value) {
        this.url$.next(value);
    }
    set sort(value) {
        this.sort$.next(value);
    }
    set filters(value) {
        this.filters$.next(value);
    }
    getOptions(params, page) {
        return this.url$.pipe(take(1), switchMap((url) => this.httpClient.get(url, {
            params: {
                ...params,
                page: page + 1,
                limit: this.pageSize,
            },
        })), map((res) => (Array.isArray(res) ? res : res?.items) ?? []));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectApiV4Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuSimpleSelectApiV4Directive, isStandalone: true, selector: "lu-simple-select[apiV4],lu-multi-select[apiV4]", inputs: { apiV4: "apiV4", sort: "sort", filters: "filters" }, usesInheritance: true, ngImport: i0 }); }
}
export { LuSimpleSelectApiV4Directive };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectApiV4Directive, decorators: [{
            type: Directive,
            args: [{
                    // The attribute is already prefixed with "lu-simple-select"
                    // eslint-disable-next-line @angular-eslint/directive-selector
                    selector: 'lu-simple-select[apiV4],lu-multi-select[apiV4]',
                    standalone: true,
                }]
        }], propDecorators: { apiV4: [{
                type: Input
            }], sort: [{
                type: Input
            }], filters: [{
                type: Input
            }] } });
