// @lucca-front/ng/esm2022/simple-select/api/api-v3.directive.mjs
import { HttpClient } from '@angular/common/http';
import { Directive, inject, Input } from '@angular/core';
import { BehaviorSubject, combineLatest, map, ReplaySubject, switchMap, take } from 'rxjs';
import { ALuSimpleSelectApiDirective } from './api.directive';
import * as i0 from "@angular/core";
class LuSimpleSelectApiV3Directive extends ALuSimpleSelectApiDirective {
    constructor() {
        super(...arguments);
        this.url$ = new ReplaySubject(1);
        this.fields$ = new BehaviorSubject('id,name');
        this.orderBy$ = new BehaviorSubject('name,asc');
        this.filters$ = new BehaviorSubject({});
        this.httpClient = inject(HttpClient);
        this.params$ = combineLatest([this.fields$, this.filters$, this.orderBy$, this.clue$]).pipe(map(([fields, filters, orderBy, clue]) => ({
            ...filters,
            fields,
            ...(orderBy ? { orderBy } : {}),
            ...(clue ? { name: `like,${clue}` } : {}),
        })));
        this.optionComparer = (a, b) => a.id === b.id;
    }
    set apiV3(value) {
        this.url$.next(value);
    }
    set fields(value) {
        this.fields$.next(value);
    }
    set orderBy(value) {
        this.orderBy$.next(value);
    }
    set filters(value) {
        this.filters$.next(value);
    }
    getOptions(params, page) {
        return this.url$.pipe(take(1), switchMap((url) => this.httpClient.get(url, {
            params: {
                ...params,
                paging: `${page * this.pageSize},${this.pageSize}`,
            },
        })), map((res) => res.data.items));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectApiV3Directive, deps: null, target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuSimpleSelectApiV3Directive, isStandalone: true, selector: "lu-simple-select[apiV3],lu-multi-select[apiV3]", inputs: { apiV3: "apiV3", fields: "fields", orderBy: "orderBy", filters: "filters" }, usesInheritance: true, ngImport: i0 }); }
}
export { LuSimpleSelectApiV3Directive };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectApiV3Directive, decorators: [{
            type: Directive,
            args: [{
                    // The attribute is already prefixed with "lu-simple-select"
                    // eslint-disable-next-line @angular-eslint/directive-selector
                    selector: 'lu-simple-select[apiV3],lu-multi-select[apiV3]',
                    standalone: true,
                }]
        }], propDecorators: { apiV3: [{
                type: Input
            }], fields: [{
                type: Input
            }], orderBy: [{
                type: Input
            }], filters: [{
                type: Input
            }] } });
