// @lucca-front/ng/esm2022/simple-select/api/api.directive.mjs
import { Directive, inject } from '@angular/core';
import { ALuSelectInputComponent } from '@lucca-front/ng/core-select';
import { BehaviorSubject, catchError, combineLatest, concatMap, debounceTime, of, scan, startWith, Subject, switchMap, takeUntil, takeWhile, tap } from 'rxjs';
import * as i0 from "@angular/core";
export const MAGIC_PAGE_SIZE = 20;
export const MAGIC_DEBOUNCE_DURATION = 250;
class ALuSimpleSelectApiDirective {
    constructor() {
        this.destroy$ = new Subject();
        this.pageSize = MAGIC_PAGE_SIZE;
        this.debounceDuration = MAGIC_DEBOUNCE_DURATION;
        this.select = inject(ALuSelectInputComponent);
        this.page$ = this.select.nextPage.pipe(scan((page) => page + 1, 0), startWith(0));
        this.clue$ = this.select.clueChange.pipe(startWith(''), debounceTime(this.debounceDuration));
        this.loading$ = new BehaviorSubject(false);
    }
    ngOnInit() {
        this.select.optionComparer = this.optionComparer;
        this.buildOptions().pipe(takeUntil(this.destroy$)).subscribe(this.select.options$);
        this.loading$.pipe(debounceTime(0), takeUntil(this.destroy$)).subscribe(this.select.loading$);
    }
    buildOptions() {
        return combineLatest([this.params$.pipe(debounceTime(0)), this.select.isPanelOpen$]).pipe(switchMap(([params, isOpened]) => isOpened
            ? this.page$.pipe(concatMap((page) => {
                this.loading$.next(true);
                return this.getOptions(params, page).pipe(catchError(() => of([])), tap(() => this.loading$.next(false)));
            }), takeWhile((items) => items.length === this.pageSize, true), scan((acc, items) => [...acc, ...items], []))
            : of([])));
    }
    ngOnDestroy() {
        this.destroy$.next();
        this.destroy$.complete();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: ALuSimpleSelectApiDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: ALuSimpleSelectApiDirective, ngImport: i0 }); }
}
export { ALuSimpleSelectApiDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: ALuSimpleSelectApiDirective, decorators: [{
            type: Directive
        }] });
