// @lucca-front/ng/esm2022/input/displayer/input-displayer.directive.mjs
import { Directive, forwardRef, Input } from '@angular/core';
import { ALuInputDisplayer } from './input-displayer.model';
import * as i0 from "@angular/core";
class LuInputDisplayerDirective extends ALuInputDisplayer {
    set argMultiple(m) {
        this.multiple = !!m;
    }
    set inputMultiple(m) {
        if (m === '') {
            // allows to have multiple = true when writing
            // <ng-template luDisplayer multiple>
            this.multiple = true;
        }
        else {
            this.multiple = !!m;
        }
    }
    constructor(template) {
        super();
        this.template = template;
    }
    getViewRef(value) {
        return this.template.createEmbeddedView({ $implicit: value });
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuInputDisplayerDirective, deps: [{ token: i0.TemplateRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuInputDisplayerDirective, isStandalone: true, selector: "[luDisplayer]", inputs: { argMultiple: ["luDisplayerMultiple", "argMultiple"], inputMultiple: ["multiple", "inputMultiple"] }, providers: [
            {
                provide: ALuInputDisplayer,
                useExisting: forwardRef(() => LuInputDisplayerDirective),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0 }); }
}
export { LuInputDisplayerDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuInputDisplayerDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luDisplayer]',
                    standalone: true,
                    providers: [
                        {
                            provide: ALuInputDisplayer,
                            useExisting: forwardRef(() => LuInputDisplayerDirective),
                            multi: true,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i0.TemplateRef }]; }, propDecorators: { argMultiple: [{
                type: Input,
                args: ['luDisplayerMultiple']
            }], inputMultiple: [{
                type: Input,
                args: ['multiple']
            }] } });
export class LuInputDisplayerContext {
    constructor($implicit) {
        this.$implicit = $implicit;
    }
}
