// @lucca-front/ng/esm2022/dropdown/panel/dropdown-panel.module.mjs
import { NgModule } from '@angular/core';
import { LuDropdownPanelComponent } from './dropdown-panel.component';
import { CommonModule } from '@angular/common';
import * as i0 from "@angular/core";
class LuDropdownPanelModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownPanelModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownPanelModule, declarations: [LuDropdownPanelComponent], imports: [CommonModule], exports: [LuDropdownPanelComponent] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownPanelModule, imports: [CommonModule] }); }
}
export { LuDropdownPanelModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownPanelModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LuDropdownPanelComponent],
                    imports: [CommonModule],
                    exports: [LuDropdownPanelComponent],
                }]
        }] });
