// @lucca-front/ng/esm2022/option/operator/pager/option-pager.component.mjs
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { ALuOnScrollBottomSubscriber } from '@lucca-front/ng/core';
import { BehaviorSubject, combineLatest } from 'rxjs';
import { map } from 'rxjs/operators';
import { ALuOptionOperator } from '../option-operator.model';
import * as i0 from "@angular/core";
const MAGIC_STEP = 10;
class LuOptionPagerComponent extends ALuOptionOperator {
    constructor() {
        super(...arguments);
        this.paging$ = new BehaviorSubject(MAGIC_STEP);
    }
    set inOptions$(in$) {
        this.outOptions$ = combineLatest([in$, this.paging$]).pipe(map(([options, paging]) => {
            return (options || []).slice(0, paging);
        }));
    }
    next() {
        this.paging$.next(this.paging$.value + MAGIC_STEP);
    }
    onScrollBottom() {
        this.next();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuOptionPagerComponent, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuOptionPagerComponent, isStandalone: true, selector: "lu-option-pager", providers: [
            {
                provide: ALuOptionOperator,
                useExisting: forwardRef(() => LuOptionPagerComponent),
                multi: true,
            },
            {
                provide: ALuOnScrollBottomSubscriber,
                useExisting: forwardRef(() => LuOptionPagerComponent),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuOptionPagerComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuOptionPagerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lu-option-pager', template: '', changeDetection: ChangeDetectionStrategy.OnPush, standalone: true, providers: [
                        {
                            provide: ALuOptionOperator,
                            useExisting: forwardRef(() => LuOptionPagerComponent),
                            multi: true,
                        },
                        {
                            provide: ALuOnScrollBottomSubscriber,
                            useExisting: forwardRef(() => LuOptionPagerComponent),
                            multi: true,
                        },
                    ] }]
        }] });
