// @lucca-front/ng/esm2022/option/operator/pager/tree-option-pager.component.mjs
import { ChangeDetectionStrategy, Component, forwardRef } from '@angular/core';
import { ALuOnScrollBottomSubscriber } from '@lucca-front/ng/core';
import { BehaviorSubject, combineLatest } from 'rxjs';
import { map } from 'rxjs/operators';
import { ALuTreeOptionOperator } from '../tree-option-operator.model';
import * as i0 from "@angular/core";
const MAGIC_STEP = 10;
class LuTreeOptionPagerComponent extends ALuTreeOptionOperator {
    constructor() {
        super(...arguments);
        this.paging$ = new BehaviorSubject(MAGIC_STEP);
    }
    set inOptions$(in$) {
        this.outOptions$ = combineLatest([in$, this.paging$]).pipe(map(([options, paging]) => {
            return this.trim(options, paging);
        }));
    }
    next() {
        this.paging$.next(this.paging$.value + MAGIC_STEP);
    }
    onScrollBottom() {
        this.next();
    }
    trim(trees = [], paging = MAGIC_STEP) {
        const flat = this.flatten(trees);
        const flatTrimmed = flat.slice(0, paging);
        return this.filter(trees, flatTrimmed);
    }
    flatten(trees = []) {
        return trees.map((t) => [t.value, ...this.flatten(t.children)]).reduce((a, v) => [...a, ...v], []);
    }
    filter(trees = [], values) {
        return trees
            .map((t) => {
            if (!values.some((v) => v === t.value)) {
                return undefined;
            }
            return { ...t, children: this.filter(t.children, values) };
        })
            .filter((t) => !!t);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTreeOptionPagerComponent, deps: null, target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuTreeOptionPagerComponent, isStandalone: true, selector: "lu-tree-option-pager", providers: [
            {
                provide: ALuTreeOptionOperator,
                useExisting: forwardRef(() => LuTreeOptionPagerComponent),
                multi: true,
            },
            {
                provide: ALuOnScrollBottomSubscriber,
                useExisting: forwardRef(() => LuTreeOptionPagerComponent),
                multi: true,
            },
        ], usesInheritance: true, ngImport: i0, template: '', isInline: true, changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
export { LuTreeOptionPagerComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTreeOptionPagerComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lu-tree-option-pager', template: '', standalone: true, changeDetection: ChangeDetectionStrategy.OnPush, providers: [
                        {
                            provide: ALuTreeOptionOperator,
                            useExisting: forwardRef(() => LuTreeOptionPagerComponent),
                            multi: true,
                        },
                        {
                            provide: ALuOnScrollBottomSubscriber,
                            useExisting: forwardRef(() => LuTreeOptionPagerComponent),
                            multi: true,
                        },
                    ] }]
        }] });
