// @lucca-front/ng/esm2022/title/title.service.mjs
import { Inject, Injectable } from '@angular/core';
import { ActivationEnd } from '@angular/router';
import { BehaviorSubject, combineLatest, of } from 'rxjs';
import { distinctUntilChanged, filter, map, switchMap, tap } from 'rxjs/operators';
import { LU_TITLE_TRANSLATE_SERVICE } from './title-translate.service';
import { TitleSeparator } from './title.model';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@angular/platform-browser";
/**
 * @deprecated use Title strategy instead
 */
class LuTitleService {
    constructor(router, title, translateService) {
        this.router = router;
        this.title = title;
        this.translateService = translateService;
        this.titlePartsSubject = new BehaviorSubject(['Lucca']);
        this.titleParts$ = this.titlePartsSubject.asObservable();
        this.title$ = this.titleParts$.pipe(switchMap((titleParts) => combineLatest(titleParts.map((part) => (typeof part === 'string' ? of(part) : part)))), map((parts) => parts.join(TitleSeparator)));
    }
    init(applicationNameTranslationKey) {
        this.router.events
            .pipe(filter((event) => {
            return event instanceof ActivationEnd && event.snapshot.children.length === 0;
        }), map((event) => getPageTitleParts(event.snapshot)), map((titleParts) => uniqTitle(titleParts)), map((titleParts) => titleParts.filter(({ title }) => title !== '').map(({ title, params }) => this.translateService.translate(title, params))), map((titleParts) => [...titleParts, this.translateService.translate(applicationNameTranslationKey, {}), 'Lucca'].filter((x) => !!x)), distinctUntilChanged(), tap((titleParts) => this.titlePartsSubject.next(titleParts)))
            .subscribe();
        this.title$.pipe(tap((title) => this.title.setTitle(title))).subscribe();
    }
    prependTitle(title) {
        this.titlePartsSubject.next([title, ...this.titlePartsSubject.value]);
    }
    overrideFirstTitlePart(title) {
        this.titlePartsSubject.next([title, ...this.titlePartsSubject.value.slice(1)]);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTitleService, deps: [{ token: i1.Router }, { token: i2.Title }, { token: LU_TITLE_TRANSLATE_SERVICE }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTitleService }); }
}
export { LuTitleService };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTitleService, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.Router }, { type: i2.Title }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [LU_TITLE_TRANSLATE_SERVICE]
                }] }]; } });
function getPageTitleParts(snapshot) {
    const pageTitle = {
        title: (snapshot.data?.['title'] || ''),
        params: { ...snapshot.params, ...snapshot.data },
    };
    return snapshot.parent ? [pageTitle, ...getPageTitleParts(snapshot.parent)] : [pageTitle];
}
function uniqTitle(titleParts) {
    return titleParts.filter(({ title }, index) => titleParts.findIndex((pageTitle) => pageTitle.title === title) === index);
}
