// @lucca-front/ng/esm2022/title/title.strategy.mjs
import { Inject, Injectable, InjectionToken } from '@angular/core';
import { TitleStrategy } from '@angular/router';
import { BehaviorSubject, combineLatest, of } from 'rxjs';
import { distinctUntilChanged, map, switchMap, tap } from 'rxjs/operators';
import { LU_TITLE_TRANSLATE_SERVICE } from './title-translate.service';
import { TitleSeparator } from './title.model';
import * as i0 from "@angular/core";
import * as i1 from "@angular/platform-browser";
export const APP_TITLE = new InjectionToken('APP_TITLE');
class LuTitleStrategy extends TitleStrategy {
    constructor(title, translateService, appTitle) {
        super();
        this.title = title;
        this.translateService = translateService;
        this.appTitle = appTitle;
        this.titlePartsSubject = new BehaviorSubject(['Lucca']);
        this.titleParts$ = this.titlePartsSubject.asObservable();
        this.title$ = this.titleParts$.pipe(switchMap((titleParts) => combineLatest(titleParts.map((part) => (typeof part === 'string' ? of(part) : part)))), map((parts) => parts.join(TitleSeparator)), distinctUntilChanged());
        this.title$.pipe(tap((title) => this.title.setTitle(title))).subscribe();
    }
    updateTitle(routerState) {
        // Title page is display from child to root
        const pageTitles = this.#getPageTitleParts(routerState.root).reverse();
        const translatedPageTitles = uniqTitle(pageTitles)
            .filter(({ title }) => title !== '')
            .map(({ title, params }) => this.translateService.translate(title, params));
        // Add the name app and 'Lucca' at the end of the title
        const titleParts = [...translatedPageTitles, this.translateService.translate(this.appTitle), 'Lucca'].filter((x) => !!x);
        this.titlePartsSubject.next(titleParts);
    }
    prependTitle(title) {
        this.titlePartsSubject.next([title, ...this.titlePartsSubject.value]);
    }
    overrideFirstTitlePart(title) {
        this.titlePartsSubject.next([title, ...this.titlePartsSubject.value.slice(1)]);
    }
    #getPageTitleParts(snapshot) {
        const pageTitle = {
            title: this.getResolvedTitleForRoute(snapshot),
            params: { ...snapshot.params, ...snapshot.data },
        };
        return snapshot.firstChild ? [pageTitle, ...this.#getPageTitleParts(snapshot.firstChild)] : [pageTitle];
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTitleStrategy, deps: [{ token: i1.Title }, { token: LU_TITLE_TRANSLATE_SERVICE }, { token: APP_TITLE }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTitleStrategy, providedIn: 'root' }); }
}
export { LuTitleStrategy };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuTitleStrategy, decorators: [{
            type: Injectable,
            args: [{ providedIn: 'root' }]
        }], ctorParameters: function () { return [{ type: i1.Title }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [LU_TITLE_TRANSLATE_SERVICE]
                }] }, { type: undefined, decorators: [{
                    type: Inject,
                    args: [APP_TITLE]
                }] }]; } });
function uniqTitle(titleParts) {
    return titleParts.filter(({ title }, index) => titleParts.findIndex((pageTitle) => pageTitle.title === title) === index);
}
