// @lucca-front/ng/esm2022/core-select/option/default-option.component.mjs
import { AsyncPipe, NgIf } from '@angular/common';
import { Component, inject } from '@angular/core';
import { LU_OPTION_CONTEXT } from './option.token';
import * as i0 from "@angular/core";
class LuSimpleSelectDefaultOptionComponent {
    constructor() {
        this.context = inject(LU_OPTION_CONTEXT);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectDefaultOptionComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuSimpleSelectDefaultOptionComponent, isStandalone: true, selector: "lu-simple-select-default-option", ngImport: i0, template: `<ng-container *ngIf="context.option$ | async as option">{{ option?.name ?? option }}</ng-container>`, isInline: true, dependencies: [{ kind: "pipe", type: AsyncPipe, name: "async" }, { kind: "directive", type: NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }] }); }
}
export { LuSimpleSelectDefaultOptionComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSimpleSelectDefaultOptionComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'lu-simple-select-default-option',
                    standalone: true,
                    imports: [AsyncPipe, NgIf],
                    template: `<ng-container *ngIf="context.option$ | async as option">{{ option?.name ?? option }}</ng-container>`,
                }]
        }] });
