// @lucca-front/ng/esm2022/core-select/option/disabled.directive.mjs
import { Directive, inject, Input } from '@angular/core';
import { LU_OPTION_CONTEXT } from './option.token';
import * as i0 from "@angular/core";
class LuDisabledOptionDirective {
    constructor() {
        this.context = inject(LU_OPTION_CONTEXT);
    }
    set isDisabled(disabled) {
        if (disabled !== null) {
            this.context.isDisabled$.next(disabled);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDisabledOptionDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuDisabledOptionDirective, isStandalone: true, selector: "[luDisabledOption]", inputs: { isDisabled: ["luDisabledOption", "isDisabled"] }, ngImport: i0 }); }
}
export { LuDisabledOptionDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDisabledOptionDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luDisabledOption]',
                    standalone: true,
                }]
        }], propDecorators: { isDisabled: [{
                type: Input,
                args: ['luDisabledOption']
            }] } });
