// @lucca-front/ng/esm2022/core-select/option/option-outlet.directive.mjs
import { Directive, inject, Injector, Input, TemplateRef, ViewContainerRef } from '@angular/core';
import { LU_OPTION_CONTEXT, provideOptionContext } from './option.token';
import * as i0 from "@angular/core";
class LuOptionOutletDirective {
    constructor() {
        this.viewContainerRef = inject(ViewContainerRef);
        this.injector = inject(Injector);
        this.optionContext = inject(LU_OPTION_CONTEXT);
    }
    ngOnChanges(changes) {
        if (changes['luOptionOutlet'] || !this.luOptionOutletValue) {
            this.clearContainer();
        }
        const hasRef = this.embeddedViewRef || this.componentRef;
        if (changes['luOptionOutlet'] || (changes['luOptionOutletValue'] && !hasRef)) {
            this.createComponent();
        }
        else if (changes['luOptionOutletValue']) {
            this.updateRefValue();
        }
    }
    ngOnDestroy() {
        this.optionContext.destroy();
    }
    clearContainer() {
        this.viewContainerRef.clear();
        this.embeddedViewRef?.destroy();
        this.componentRef?.destroy();
        this.embeddedViewRef = undefined;
        this.componentRef = undefined;
    }
    createComponent() {
        if (!this.luOptionOutlet || !this.luOptionOutletValue) {
            return;
        }
        if (this.luOptionOutlet instanceof TemplateRef) {
            this.embeddedViewRef = this.viewContainerRef.createEmbeddedView(this.luOptionOutlet, { $implicit: this.luOptionOutletValue }, { injector: this.injector });
        }
        else {
            this.optionContext.option$.next(this.luOptionOutletValue);
            this.componentRef = this.viewContainerRef.createComponent(this.luOptionOutlet, { injector: this.injector });
        }
    }
    updateRefValue() {
        if (!this.luOptionOutletValue) {
            return;
        }
        if (this.embeddedViewRef) {
            this.embeddedViewRef.context = { $implicit: this.luOptionOutletValue };
        }
        else if (this.componentRef) {
            this.optionContext.option$.next(this.luOptionOutletValue);
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuOptionOutletDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuOptionOutletDirective, isStandalone: true, selector: "[luOptionOutlet]", inputs: { luOptionOutlet: "luOptionOutlet", luOptionOutletValue: "luOptionOutletValue" }, providers: [provideOptionContext()], usesOnChanges: true, ngImport: i0 }); }
}
export { LuOptionOutletDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuOptionOutletDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luOptionOutlet]',
                    standalone: true,
                    providers: [provideOptionContext()],
                }]
        }], propDecorators: { luOptionOutlet: [{
                type: Input
            }], luOptionOutletValue: [{
                type: Input
            }] } });
