// @lucca-front/ng/esm2022/dropdown/item/dropdown-item.module.mjs
import { NgModule } from '@angular/core';
import { LuDropdownItemDirective } from './dropdown-item.directive';
import * as i0 from "@angular/core";
class LuDropdownItemModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownItemModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownItemModule, declarations: [LuDropdownItemDirective], exports: [LuDropdownItemDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownItemModule }); }
}
export { LuDropdownItemModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownItemModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [LuDropdownItemDirective],
                    imports: [],
                    exports: [LuDropdownItemDirective],
                }]
        }] });
