// @lucca-front/ng/esm2022/dropdown/item/dropdown-item.directive.mjs
import { Directive, EventEmitter, forwardRef, HostListener, Output } from '@angular/core';
import { ALuDropdownItem } from './dropdown-item.model';
import * as i0 from "@angular/core";
class LuDropdownItemDirective extends ALuDropdownItem {
    constructor(_eltRef) {
        super();
        this._eltRef = _eltRef;
        // arreter les ONxxxx
        // eslint-disable-next-line @angular-eslint/no-output-on-prefix
        this.onSelect = new EventEmitter();
    }
    onClick() {
        this.onSelect.emit(true);
    }
    onEnter() {
        this.onSelect.emit(true);
    }
    focus() {
        this._eltRef.nativeElement.focus();
    }
    ngOnDestroy() {
        this.onSelect.complete();
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownItemDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuDropdownItemDirective, selector: "[luDropdownItem]", outputs: { onSelect: "onSelect" }, host: { listeners: { "click": "onClick($event)", "keydown.enter": "onEnter($event)" } }, providers: [
            {
                provide: ALuDropdownItem,
                useExisting: forwardRef(() => LuDropdownItemDirective),
                multi: true,
            },
        ], exportAs: ["LuDropdownItem"], usesInheritance: true, ngImport: i0 }); }
}
export { LuDropdownItemDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDropdownItemDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luDropdownItem]',
                    exportAs: 'LuDropdownItem',
                    providers: [
                        {
                            provide: ALuDropdownItem,
                            useExisting: forwardRef(() => LuDropdownItemDirective),
                            multi: true,
                        },
                    ],
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { onSelect: [{
                type: Output
            }], onClick: [{
                type: HostListener,
                args: ['click', ['$event']]
            }], onEnter: [{
                type: HostListener,
                args: ['keydown.enter', ['$event']]
            }] } });
