// @lucca-front/ng/esm2022/forms/abstract-field-component.mjs
import { booleanAttribute, Component, inject } from '@angular/core';
import { injectNgControl } from './inject-ng-control';
import { NG_VALIDATORS, RequiredValidator, Validators } from '@angular/forms';
import * as i0 from "@angular/core";
class AbstractFieldComponent {
    constructor() {
        this.ngControl = injectNgControl();
        this.#ngModelRequiredValidator = inject(NG_VALIDATORS, { optional: true })?.find((v) => v instanceof RequiredValidator);
    }
    #ngModelRequiredValidator;
    get required() {
        return this.ngControl.control.hasValidator(Validators.required) || booleanAttribute(this.#ngModelRequiredValidator.required);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: AbstractFieldComponent, deps: [], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: AbstractFieldComponent, isStandalone: true, selector: "ng-component", ngImport: i0, template: '', isInline: true }); }
}
export { AbstractFieldComponent };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: AbstractFieldComponent, decorators: [{
            type: Component,
            args: [{
                    template: '',
                    standalone: true,
                }]
        }] });
