// @lucca-front/ng/esm2022/forms/form-field-id.directive.mjs
import { Directive, HostBinding, inject, Input } from '@angular/core';
import { FormFieldComponent } from '@lucca-front/ng/form-field';
import { filter, take } from 'rxjs/operators';
import * as i0 from "@angular/core";
class FormFieldIdDirective {
    #formFieldComponent;
    #suffix;
    set suffix(suffix) {
        this.#suffix = suffix;
        if (this.#formFieldComponent.ready) {
            this.applyLabelledBy();
        }
    }
    get id() {
        return `${this.#formFieldComponent.id}-${this.#suffix}`;
    }
    constructor() {
        this.#formFieldComponent = inject(FormFieldComponent);
        this.labelledByStrategy = 'append';
        // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
        this.#formFieldComponent.ready$.pipe(filter(Boolean), take(1)).subscribe(() => {
            this.applyLabelledBy();
        });
    }
    applyLabelledBy() {
        this.#formFieldComponent.addLabelledBy(`${this.#formFieldComponent.id}-${this.#suffix}`, this.labelledByStrategy === 'prepend');
    }
    ngOnDestroy() {
        this.#formFieldComponent.removeLabelledBy(`${this.#formFieldComponent.id}-${this.#suffix}`);
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: FormFieldIdDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: FormFieldIdDirective, isStandalone: true, selector: "[luFormFieldId]", inputs: { suffix: ["luFormFieldId", "suffix"], labelledByStrategy: "labelledByStrategy" }, host: { properties: { "attr.id": "this.id" } }, ngImport: i0 }); }
}
export { FormFieldIdDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: FormFieldIdDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luFormFieldId]',
                    standalone: true,
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { suffix: [{
                type: Input,
                args: [{
                        required: true,
                        alias: 'luFormFieldId',
                    }]
            }], labelledByStrategy: [{
                type: Input
            }], id: [{
                type: HostBinding,
                args: ['attr.id']
            }] } });
