// @lucca-front/ng/esm2022/forms/noop-value-accessor.directive.mjs
//Source: https://netbasal.com/forwarding-form-controls-to-custom-control-components-in-angular-701e8406cc55
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { Directive } from '@angular/core';
import * as i0 from "@angular/core";
class NoopValueAccessorDirective {
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    writeValue() { }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    registerOnChange() { }
    // eslint-disable-next-line @typescript-eslint/no-empty-function
    registerOnTouched() { }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: NoopValueAccessorDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: NoopValueAccessorDirective, isStandalone: true, selector: "[luNoopValueAccessor]", providers: [
            {
                provide: NG_VALUE_ACCESSOR,
                multi: true,
                useExisting: NoopValueAccessorDirective,
            },
        ], ngImport: i0 }); }
}
export { NoopValueAccessorDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: NoopValueAccessorDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luNoopValueAccessor]',
                    standalone: true,
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            multi: true,
                            useExisting: NoopValueAccessorDirective,
                        },
                    ],
                }]
        }] });
