// @lucca-front/ng/esm2022/multi-select/panel/selected-chip.directive.mjs
import { Directive, ElementRef, Input, inject } from '@angular/core';
import * as i0 from "@angular/core";
class ɵLuMultiSelectSelectedChipDirective {
    constructor() {
        this.elementRef = inject(ElementRef).nativeElement;
    }
    setActiveStyles() {
        this.elementRef.querySelector('button').focus();
    }
    setInactiveStyles() {
        // Nothing to do
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: ɵLuMultiSelectSelectedChipDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: ɵLuMultiSelectSelectedChipDirective, isStandalone: true, selector: "[luMultiSelectSelectedChip]", inputs: { option: ["luMultiSelectSelectedChip", "option"] }, ngImport: i0 }); }
}
export { ɵLuMultiSelectSelectedChipDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: ɵLuMultiSelectSelectedChipDirective, decorators: [{
            type: Directive,
            args: [{
                    standalone: true,
                    selector: '[luMultiSelectSelectedChip]',
                }]
        }], propDecorators: { option: [{
                type: Input,
                args: ['luMultiSelectSelectedChip']
            }] } });
