// @lucca-front/ng/esm2022/department/service/department-v3.service.mjs
import { Injectable } from '@angular/core';
import { LuApiV3Service } from '@lucca-front/ng/api';
import { map } from 'rxjs/operators';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
class LuDepartmentV3Service extends LuApiV3Service {
    set appInstanceId(appInstanceId) {
        if (appInstanceId) {
            this._appInstanceId = appInstanceId;
        }
    }
    set operations(operations) {
        this._operations = operations;
    }
    constructor(_http) {
        super(_http);
        this._http = _http;
        this._api = `/api/v3/departments`;
        this._operations = [];
    }
    getTrees() {
        let call;
        if (this._appInstanceId && this._operations?.length) {
            call = this._http.get(`/api/v3/departments/scopedtree?fields=id,name&${[`appInstanceId=${this._appInstanceId}`, `operations=${this._operations.join(',')}`, this._filters.join(',')].filter((f) => !!f).join('&')}`);
        }
        else {
            call = this._http.get(`/api/v3/departments/tree?fields=id,name&${this._filters.join(',')}`);
        }
        return call.pipe(map((response) => {
            const tree = response.data;
            return tree?.children.map((c) => this.format(c)) ?? [];
        }));
    }
    format(t) {
        return { value: t.node, children: t.children.map((c) => this.format(c)) };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDepartmentV3Service, deps: [{ token: i1.HttpClient }], target: i0.ɵɵFactoryTarget.Injectable }); }
    static { this.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDepartmentV3Service }); }
}
export { LuDepartmentV3Service };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuDepartmentV3Service, decorators: [{
            type: Injectable
        }], ctorParameters: function () { return [{ type: i1.HttpClient }]; } });
