// @lucca-front/ng/esm2022/safe-content/safe-content.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LuSafeHtmlPipe } from './safe-html.pipe';
import * as i0 from "@angular/core";
class LuSafeContentModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSafeContentModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuSafeContentModule, declarations: [LuSafeHtmlPipe], imports: [CommonModule], exports: [LuSafeHtmlPipe] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSafeContentModule, imports: [CommonModule] }); }
}
export { LuSafeContentModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSafeContentModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [LuSafeHtmlPipe],
                    exports: [LuSafeHtmlPipe],
                    providers: [],
                }]
        }] });
