// @lucca-front/ng/fesm2022/lucca-front-ng-number.mjs
import * as i0 from '@angular/core';
import { LOCALE_ID, Pipe, Inject, NgModule } from '@angular/core';
import { formatNumber, getLocaleNumberSymbol, NumberSymbol } from '@angular/common';

class LuNumberPipe {
    constructor(locale) {
        this.locale = locale;
    }
    transform(number, precision = 2) {
        const formatted = formatNumber(number, this.locale, `1.${precision}-${precision}`);
        const separator = getLocaleNumberSymbol(this.locale, NumberSymbol.Decimal);
        const split = formatted.split(separator);
        const integral = split[0];
        const decimal = split[1];
        const hideDecimal = Math.round(number) === number;
        if (precision > 0) {
            return `${integral}<span class="decimal-part${hideDecimal ? ' u-hidden' : ''}">${separator}${decimal}</span>`;
        }
        else {
            return integral;
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberPipe, deps: [{ token: LOCALE_ID }], target: i0.ɵɵFactoryTarget.Pipe }); }
    static { this.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuNumberPipe, isStandalone: true, name: "luNumber" }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'luNumber',
                    pure: true,
                    standalone: true,
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [LOCALE_ID]
                }] }]; } });

class LuNumberModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuNumberModule, imports: [LuNumberPipe], exports: [LuNumberPipe] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuNumberModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [LuNumberPipe],
                    exports: [LuNumberPipe],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { LuNumberModule, LuNumberPipe };
