// @lucca-front/ng/fesm2022/lucca-front-ng-a11y.mjs
import { DOCUMENT } from '@angular/common';
import * as i0 from '@angular/core';
import { InjectionToken, Component, ChangeDetectionStrategy, Inject } from '@angular/core';
import { getIntl } from '@lucca-front/ng/core';

const LU_SKIP_LINKS_TRANSLATIONS = new InjectionToken('LuModalTranslations', {
    factory: () => luSkipLinksTranslations,
});
class ALuSkipLinksLabel {
}
const luSkipLinksTranslations = {
    en: {
        Goto: 'Quick access',
        Goto_Nav_Banner: 'Go to main navigation',
        Goto_Nav_Navside: 'Go to internal navigation',
        Goto_Content: 'Go to content',
    },
    fr: {
        Goto: 'Accès rapides',
        Goto_Nav_Banner: 'Aller à la navigation principale',
        Goto_Nav_Navside: 'Aller à la navigation interne',
        Goto_Content: 'Aller au contenu',
    },
    es: {
        Goto: 'Acceso rápido',
        Goto_Nav_Banner: 'Ir a la navegación principal',
        Goto_Nav_Navside: 'Ir a la navegación interna',
        Goto_Content: 'Ir al contenido',
    },
};

class LuSkipLinksComponent {
    constructor(document) {
        this.document = document;
        this.intl = getIntl(LU_SKIP_LINKS_TRANSLATIONS);
    }
    anchor(hash, e) {
        e.preventDefault();
        this.document.location.hash = '';
        this.document.location.hash = hash;
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSkipLinksComponent, deps: [{ token: DOCUMENT }], target: i0.ɵɵFactoryTarget.Component }); }
    static { this.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "16.1.3", type: LuSkipLinksComponent, isStandalone: true, selector: "lu-skip-links", ngImport: i0, template: "<nav role=\"navigation\" id=\"top\" class=\"skipLinks\" [attr.aria-label]=\"intl.Goto\">\n\t<a\n\t\tclass=\"skipLinks-action palette-grey mod-XS\"\n\t\thref=\"#lucca-banner-solutions-container\"\n\t\t(click)=\"anchor('#lucca-banner-solutions-container', $event)\"\n\t\t>{{ intl.Goto_Nav_Banner }}</a\n\t>\n\t<a class=\"skipLinks-action palette-grey mod-XS\" href=\"#navSide\" (click)=\"anchor('#navSide', $event)\">{{ intl.Goto_Nav_Navside }}</a>\n\t<a class=\"skipLinks-action palette-grey mod-XS\" href=\"#main-content\" (click)=\"anchor('#main-content', $event)\">{{ intl.Goto_Content }}</a>\n</nav>\n", styles: [".skipLinks{position:fixed;list-style-type:none;justify-content:center;display:flex;margin:0;padding:0;z-index:9010;inset:calc(var(--spacings-XS) + var(--spacings-XXS));bottom:auto}.skipLinks-action{border:0;margin:0;font-weight:600;border-radius:var(--commons-borderRadius-M);font-size:var(--sizes-XS-fontSize);line-height:var(--sizes-XS-lineHeight);padding:var(--spacings-XXS) var(--spacings-XS);transition:all var(--commons-animations-durations-fast);background-color:var(--palettes-grey-600);display:inline-block;position:relative;text-decoration:none;vertical-align:baseline;white-space:nowrap;color:var(--colors-white-color);cursor:pointer}.skipLinks-action:focus-visible{outline-offset:2px;outline:2px solid var(--palettes-primary-700)}.skipLinks-action:not(:focus,:focus-visible){border:0;clip:rect(1px,1px,1px,1px);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;width:1px;white-space:nowrap;contain:paint}::ng-deep .navSide,::ng-deep .main-content{scroll-margin-top:var(--commons-banner-height)}\n"], changeDetection: i0.ChangeDetectionStrategy.OnPush }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuSkipLinksComponent, decorators: [{
            type: Component,
            args: [{ selector: 'lu-skip-links', changeDetection: ChangeDetectionStrategy.OnPush, standalone: true, template: "<nav role=\"navigation\" id=\"top\" class=\"skipLinks\" [attr.aria-label]=\"intl.Goto\">\n\t<a\n\t\tclass=\"skipLinks-action palette-grey mod-XS\"\n\t\thref=\"#lucca-banner-solutions-container\"\n\t\t(click)=\"anchor('#lucca-banner-solutions-container', $event)\"\n\t\t>{{ intl.Goto_Nav_Banner }}</a\n\t>\n\t<a class=\"skipLinks-action palette-grey mod-XS\" href=\"#navSide\" (click)=\"anchor('#navSide', $event)\">{{ intl.Goto_Nav_Navside }}</a>\n\t<a class=\"skipLinks-action palette-grey mod-XS\" href=\"#main-content\" (click)=\"anchor('#main-content', $event)\">{{ intl.Goto_Content }}</a>\n</nav>\n", styles: [".skipLinks{position:fixed;list-style-type:none;justify-content:center;display:flex;margin:0;padding:0;z-index:9010;inset:calc(var(--spacings-XS) + var(--spacings-XXS));bottom:auto}.skipLinks-action{border:0;margin:0;font-weight:600;border-radius:var(--commons-borderRadius-M);font-size:var(--sizes-XS-fontSize);line-height:var(--sizes-XS-lineHeight);padding:var(--spacings-XXS) var(--spacings-XS);transition:all var(--commons-animations-durations-fast);background-color:var(--palettes-grey-600);display:inline-block;position:relative;text-decoration:none;vertical-align:baseline;white-space:nowrap;color:var(--colors-white-color);cursor:pointer}.skipLinks-action:focus-visible{outline-offset:2px;outline:2px solid var(--palettes-primary-700)}.skipLinks-action:not(:focus,:focus-visible){border:0;clip:rect(1px,1px,1px,1px);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;width:1px;white-space:nowrap;contain:paint}::ng-deep .navSide,::ng-deep .main-content{scroll-margin-top:var(--commons-banner-height)}\n"] }]
        }], ctorParameters: function () { return [{ type: Document, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }]; } });

/**
 * Generated bundle index. Do not edit.
 */

export { LuSkipLinksComponent };
