// @lucca-front/ng/fesm2022/lucca-front-ng-scroll.mjs
import * as i0 from '@angular/core';
import { EventEmitter, Directive, Input, Output, HostListener, NgModule } from '@angular/core';
import { Subject } from 'rxjs';
import { debounceTime } from 'rxjs/operators';

/* eslint-disable @angular-eslint/no-output-on-prefix */
/**
 * emits on scroll events
 */
class LuScrollDirective {
    _scroll($event) {
        this.scrollSubject.next($event);
    }
    ngOnInit() {
        this.scroll$.subscribe((scrollEvent) => this.emitScrollEvents(scrollEvent));
    }
    emitScrollEvents($event) {
        this.onScroll.emit($event);
        const target = $event.target;
        if (target instanceof Element) {
            const scrollLeft = target.scrollLeft;
            const scrollTop = target.scrollTop;
            if (scrollLeft === 0) {
                this.onScrollLeft.emit($event);
            }
            if (scrollTop === 0) {
                this.onScrollTop.emit($event);
            }
            const scrollWidth = target.scrollWidth;
            const scrollHeight = target.scrollHeight;
            const eltHeight = this._elementRef.nativeElement.clientHeight;
            const eltWidth = this._elementRef.nativeElement.clientWidth;
            if (scrollWidth - scrollLeft - eltWidth < 10) {
                this.onScrollRight.emit($event);
            }
            if (scrollHeight - scrollTop - eltHeight < 10) {
                this.onScrollBottom.emit($event);
            }
        }
    }
    constructor(_elementRef) {
        this._elementRef = _elementRef;
        this.debounceTime = 100;
        this.onScroll = new EventEmitter();
        this.onScrollTop = new EventEmitter();
        this.onScrollBottom = new EventEmitter();
        this.onScrollLeft = new EventEmitter();
        this.onScrollRight = new EventEmitter();
        this.scrollSubject = new Subject();
        this.scroll$ = this.scrollSubject.asObservable().pipe(debounceTime(this.debounceTime));
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuScrollDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: LuScrollDirective, isStandalone: true, selector: "[luScroll]", inputs: { debounceTime: "debounceTime" }, outputs: { onScroll: "onScroll", onScrollTop: "onScrollTop", onScrollBottom: "onScrollBottom", onScrollLeft: "onScrollLeft", onScrollRight: "onScrollRight" }, host: { listeners: { "scroll": "_scroll($event)" } }, exportAs: ["luScroll"], ngImport: i0 }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuScrollDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luScroll]',
                    exportAs: 'luScroll',
                    standalone: true,
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; }, propDecorators: { debounceTime: [{
                type: Input
            }], onScroll: [{
                type: Output
            }], onScrollTop: [{
                type: Output
            }], onScrollBottom: [{
                type: Output
            }], onScrollLeft: [{
                type: Output
            }], onScrollRight: [{
                type: Output
            }], _scroll: [{
                type: HostListener,
                args: ['scroll', ['$event']]
            }] } });

class LuScrollModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuScrollModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuScrollModule, imports: [LuScrollDirective], exports: [LuScrollDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuScrollModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuScrollModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [LuScrollDirective],
                    exports: [LuScrollDirective],
                }]
        }] });

/**
 * Generated bundle index. Do not edit.
 */

export { LuScrollDirective, LuScrollModule };
