// @lucca-front/ng/esm2022/core/portal/portal.directive.mjs
import { Directive, inject, Input, Renderer2, TemplateRef, ViewContainerRef } from '@angular/core';
import * as i0 from "@angular/core";
class PortalDirective {
    constructor() {
        this.viewContainerRef = inject(ViewContainerRef);
        this.renderer = inject(Renderer2);
        this.templateRef = inject(TemplateRef);
        this.luPortal = null;
    }
    render() {
        this.viewContainerRef.clear();
        if (this.luPortal instanceof TemplateRef) {
            this.viewContainerRef.createEmbeddedView(this.luPortal);
        }
        else {
            const ref = this.viewContainerRef.createEmbeddedView(this.templateRef);
            const container = ref.rootNodes[0];
            const parent = container.parentElement;
            this.renderer.insertBefore(parent, document.createTextNode(this.luPortal), container);
        }
    }
    ngOnChanges(changes) {
        if (changes['luPortal']) {
            // If we're here, it means that either the template ref changed or the string changed,
            // meaning that we need to render again
            this.render();
        }
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: PortalDirective, deps: [], target: i0.ɵɵFactoryTarget.Directive }); }
    static { this.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "16.1.3", type: PortalDirective, isStandalone: true, selector: "[luPortal]", inputs: { luPortal: "luPortal" }, usesOnChanges: true, ngImport: i0 }); }
}
export { PortalDirective };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: PortalDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[luPortal]',
                    standalone: true,
                }]
        }], propDecorators: { luPortal: [{
                type: Input,
                args: [{ required: true }]
            }] } });
