// @lucca-front/ng/esm2022/popup/popup.module.mjs
import { NgModule } from '@angular/core';
import { OverlayModule } from '@angular/cdk/overlay';
import { LuPopup } from './popup.service';
import { LU_POPUP_CONFIG, LU_POPUP_REF_FACTORY } from './popup.token';
import { luDefaultPopupConfig } from './popup-config.default';
import { LuPopupRefFactory } from './popup-ref.factory';
import * as i0 from "@angular/core";
class LuPopupModule {
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuPopupModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "16.1.3", ngImport: i0, type: LuPopupModule, imports: [OverlayModule] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuPopupModule, providers: [LuPopup, { provide: LU_POPUP_CONFIG, useValue: luDefaultPopupConfig }, { provide: LU_POPUP_REF_FACTORY, useClass: LuPopupRefFactory }], imports: [OverlayModule] }); }
}
export { LuPopupModule };
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "16.1.3", ngImport: i0, type: LuPopupModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [OverlayModule],
                    exports: [],
                    providers: [LuPopup, { provide: LU_POPUP_CONFIG, useValue: luDefaultPopupConfig }, { provide: LU_POPUP_REF_FACTORY, useClass: LuPopupRefFactory }],
                }]
        }] });
